// @catppuccin/vscode@3.18.0/dist/compile.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/compile.ts
var compile_exports = {};
__export(compile_exports, {
  compile: () => compile
});
module.exports = __toCommonJS(compile_exports);

// ../catppuccin-vsc/src/theme/index.ts
var import_palette2 = require("@catppuccin/palette");

// ../catppuccin-vsc/src/theme/tokens/cpp.ts
var tokens = (context) => {
  const { palette } = context;
  return [
    {
      name: "C++ extern keyword",
      scope: "storage.modifier.specifier.extern.cpp",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "C++ scope resolution",
      scope: [
        "entity.name.scope-resolution.template.call.cpp",
        "entity.name.scope-resolution.parameter.cpp",
        "entity.name.scope-resolution.cpp",
        "entity.name.scope-resolution.function.definition.cpp"
      ],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "C++ doc keywords",
      scope: "storage.type.class.doxygen",
      settings: {
        fontStyle: ""
      }
    },
    {
      name: "C++ operators",
      scope: ["storage.modifier.reference.cpp"],
      settings: {
        foreground: palette.teal
      }
    }
  ];
};
var cpp_default = tokens;

// ../catppuccin-vsc/src/theme/tokens/cs.ts
var tokens2 = (context) => {
  const { palette } = context;
  return [
    {
      name: "C# Interpolated Strings",
      scope: "meta.interpolation.cs",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "C# xml-style docs",
      scope: "comment.block.documentation.cs",
      settings: {
        foreground: palette.text
      }
    }
  ];
};
var cs_default = tokens2;

// ../catppuccin-vsc/src/theme/tokens/css.ts
var tokens3 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Classes, reflecting the className color in JSX",
      scope: [
        "source.css entity.other.attribute-name.class.css",
        "entity.other.attribute-name.parent-selector.css punctuation.definition.entity.css"
      ],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Operators",
      scope: "punctuation.separator.operator.css",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Pseudo classes",
      scope: "source.css entity.other.attribute-name.pseudo-class",
      settings: {
        foreground: palette.teal
      }
    },
    {
      scope: "source.css constant.other.unicode-range",
      settings: {
        foreground: palette.peach
      }
    },
    {
      scope: "source.css variable.parameter.url",
      settings: {
        foreground: palette.green,
        fontStyle: ""
      }
    },
    {
      name: "CSS vendored property names",
      scope: ["support.type.vendored.property-name"],
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "Less/SCSS right-hand variables (@/$-prefixed)",
      scope: [
        "source.css meta.property-value variable",
        "source.css meta.property-value variable.other.less",
        "source.css meta.property-value variable.other.less punctuation.definition.variable.less",
        // the left hand equivalent, initial assignment
        "meta.definition.variable.scss"
        // "meta.property-list variable.scss",
      ],
      settings: {
        foreground: palette.maroon
      }
    },
    {
      name: "CSS variables (--prefixed)",
      scope: [
        "source.css meta.property-list variable",
        // ditto for less
        "meta.property-list variable.other.less",
        "meta.property-list variable.other.less punctuation.definition.variable.less"
      ],
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "CSS Percentage values, styled the same as numbers",
      scope: "keyword.other.unit.percentage.css",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "CSS Attribute selectors, styled the same as strings",
      scope: "source.css meta.attribute-selector",
      settings: {
        foreground: palette.green
      }
    }
  ];
};
var css_default = tokens3;

// ../catppuccin-vsc/src/theme/tokens/data.ts
var tokens4 = (context) => {
  const { palette } = context;
  return [
    {
      name: "JSON/YAML keys, other left-hand assignments",
      scope: [
        "keyword.other.definition.ini",
        "punctuation.support.type.property-name.json",
        "support.type.property-name.json",
        "punctuation.support.type.property-name.toml",
        "support.type.property-name.toml",
        "entity.name.tag.yaml",
        "punctuation.support.type.property-name.yaml",
        "support.type.property-name.yaml"
      ],
      settings: {
        foreground: palette.blue,
        fontStyle: ""
      }
    },
    {
      name: "JSON/YAML constants",
      scope: ["constant.language.json", "constant.language.yaml"],
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "YAML anchors",
      scope: ["entity.name.type.anchor.yaml", "variable.other.alias.yaml"],
      settings: {
        foreground: palette.yellow,
        fontStyle: ""
      }
    },
    {
      name: "TOML tables / ini groups",
      scope: [
        "support.type.property-name.table",
        "entity.name.section.group-title.ini"
      ],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "TOML dates",
      scope: "constant.other.time.datetime.offset.toml",
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "YAML anchor puctuation",
      scope: [
        "punctuation.definition.anchor.yaml",
        "punctuation.definition.alias.yaml"
      ],
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "YAML triple dashes",
      scope: "entity.other.document.begin.yaml",
      settings: {
        foreground: palette.pink
      }
    }
  ];
};
var data_default = tokens4;

// ../catppuccin-vsc/src/theme/tokens/diff.ts
var tokens5 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Markup Diff",
      scope: "markup.changed.diff",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Diff",
      scope: [
        "meta.diff.header.from-file",
        "meta.diff.header.to-file",
        "punctuation.definition.from-file.diff",
        "punctuation.definition.to-file.diff"
      ],
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "Diff Inserted",
      scope: "markup.inserted.diff",
      settings: {
        foreground: palette.green
      }
    },
    {
      name: "Diff Deleted",
      scope: "markup.deleted.diff",
      settings: {
        foreground: palette.red
      }
    }
  ];
};
var diff_default = tokens5;

// ../catppuccin-vsc/src/theme/tokens/dotenv.ts
var tokens6 = (context) => {
  const { palette } = context;
  return [
    {
      name: "dotenv left-hand side assignments",
      scope: ["variable.other.env"],
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "dotenv reference to existing env variable",
      scope: ["string.quoted variable.other.env"],
      settings: {
        foreground: palette.text
      }
    }
  ];
};
var dotenv_default = tokens6;

// ../catppuccin-vsc/src/theme/tokens/gdscript.ts
var tokens7 = (context) => {
  const { palette } = context;
  return [
    {
      name: "GDScript functions",
      scope: "support.function.builtin.gdscript",
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "GDScript constants",
      scope: "constant.language.gdscript",
      settings: {
        foreground: palette.peach
      }
    }
  ];
};
var gdscript_default = tokens7;

// ../catppuccin-vsc/src/theme/tokens/golang.ts
var tokens8 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Comment keywords",
      scope: "comment meta.annotation.go",
      settings: {
        foreground: palette.maroon
      }
    },
    {
      name: "go:embed, go:build, etc.",
      scope: "comment meta.annotation.parameters.go",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Go constants (nil, true, false)",
      // nil should be peach, but TextMate doesn't offer a fine-grained enough scope
      scope: "constant.language.go",
      settings: {
        foreground: palette.peach
      }
    }
  ];
};
var golang_default = tokens8;

// ../catppuccin-vsc/src/theme/tokens/graphql.ts
var tokens9 = (context) => {
  const { palette } = context;
  return [
    {
      name: "GraphQL variables",
      scope: "variable.graphql",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "GraphQL aliases",
      scope: "string.unquoted.alias.graphql",
      settings: {
        foreground: palette.flamingo
      }
    },
    {
      name: "GraphQL enum members",
      scope: "constant.character.enum.graphql",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "GraphQL field in types",
      scope: "meta.objectvalues.graphql constant.object.key.graphql string.unquoted.graphql",
      settings: {
        foreground: palette.flamingo
      }
    }
  ];
};
var graphql_default = tokens9;

// ../catppuccin-vsc/src/theme/tokens/html.ts
var tokens10 = (context) => {
  const { palette } = context;
  return [
    {
      name: "HTML/XML DOCTYPE as keyword",
      scope: [
        "keyword.other.doctype",
        "meta.tag.sgml.doctype punctuation.definition.tag",
        "meta.tag.metadata.doctype entity.name.tag",
        "meta.tag.metadata.doctype punctuation.definition.tag"
      ],
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "HTML/XML-like <tags/>",
      scope: ["entity.name.tag"],
      settings: {
        foreground: palette.blue,
        fontStyle: ""
      }
    },
    {
      name: "Special characters like &amp;",
      scope: [
        "text.html constant.character.entity",
        "text.html constant.character.entity punctuation",
        "constant.character.entity.xml",
        "constant.character.entity.xml punctuation",
        // ditto for JSX / TSX
        "constant.character.entity.js.jsx",
        "constant.charactger.entity.js.jsx punctuation",
        "constant.character.entity.tsx",
        "constant.character.entity.tsx punctuation"
      ],
      settings: {
        foreground: palette.red
      }
    },
    {
      name: "HTML/XML tag attribute values",
      scope: ["entity.other.attribute-name"],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Components",
      scope: [
        "support.class.component",
        "support.class.component.jsx",
        "support.class.component.tsx",
        "support.class.component.vue"
      ],
      settings: {
        foreground: palette.pink,
        fontStyle: ""
      }
    }
  ];
};
var html_default = tokens10;

// ../catppuccin-vsc/src/theme/tokens/java.ts
var tokens11 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Annotations",
      scope: ["punctuation.definition.annotation", "storage.type.annotation"],
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Java enums",
      scope: "constant.other.enum.java",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Java imports",
      scope: "storage.modifier.import.java",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Javadoc",
      scope: "comment.block.javadoc.java keyword.other.documentation.javadoc.java",
      settings: { fontStyle: "" }
    }
  ];
};
var java_default = tokens11;

// ../catppuccin-vsc/src/theme/tokens/javascript.ts
var tokens12 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Exported Variable",
      scope: "meta.export variable.other.readwrite.js",
      settings: {
        foreground: palette.maroon
      }
    },
    {
      name: "JS/TS constants & properties",
      scope: [
        "variable.other.constant.js",
        "variable.other.constant.ts",
        "variable.other.property.js",
        "variable.other.property.ts"
      ],
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "JSDoc; these are mainly params, so styled as such",
      scope: [
        "variable.other.jsdoc",
        "comment.block.documentation variable.other"
      ],
      settings: {
        foreground: palette.maroon,
        fontStyle: ""
      }
    },
    {
      name: "JSDoc keywords",
      scope: "storage.type.class.jsdoc",
      settings: {
        fontStyle: ""
      }
    },
    {
      scope: "support.type.object.console.js",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Node constants as keywords (module, etc.)",
      scope: ["support.constant.node", "support.type.object.module.js"],
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "implements as keyword",
      scope: "storage.modifier.implements",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "Builtin types",
      scope: [
        "constant.language.null.js",
        "constant.language.null.ts",
        "constant.language.undefined.js",
        "constant.language.undefined.ts",
        "support.type.builtin.ts"
      ],
      settings: {
        foreground: palette.mauve
      }
    },
    {
      scope: "variable.parameter.generic",
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Arrow functions",
      scope: [
        "keyword.declaration.function.arrow.js",
        "storage.type.function.arrow.ts"
      ],
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Decorator punctuations (decorators inherit from blue functions, instead of styleguide peach)",
      scope: "punctuation.decorator.ts",
      settings: {
        foreground: palette.blue,
        fontStyle: "italic"
      }
    },
    {
      name: "Extra JS/TS keywords",
      scope: [
        "keyword.operator.expression.in.js",
        "keyword.operator.expression.in.ts",
        "keyword.operator.expression.infer.ts",
        "keyword.operator.expression.instanceof.js",
        "keyword.operator.expression.instanceof.ts",
        "keyword.operator.expression.is",
        "keyword.operator.expression.keyof.ts",
        "keyword.operator.expression.of.js",
        "keyword.operator.expression.of.ts",
        "keyword.operator.expression.typeof.ts"
      ],
      settings: {
        foreground: palette.mauve
      }
    }
  ];
};
var javascript_default = tokens12;

// ../catppuccin-vsc/src/theme/tokens/julia.ts
var tokens13 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Julia macros",
      scope: "support.function.macro.julia",
      settings: {
        foreground: palette.teal,
        fontStyle: "italic"
      }
    },
    {
      name: "Julia language constants (true, false)",
      scope: "constant.language.julia",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Julia other constants (these seem to be arguments inside arrays)",
      scope: "constant.other.symbol.julia",
      settings: {
        foreground: palette.maroon
      }
    }
  ];
};
var julia_default = tokens13;

// ../catppuccin-vsc/src/theme/tokens/latex.ts
var tokens14 = (context) => {
  const { palette } = context;
  return [
    {
      name: "LaTeX preamble",
      scope: "text.tex keyword.control.preamble",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "LaTeX be functions",
      scope: "text.tex support.function.be",
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "LaTeX math",
      scope: "constant.other.general.math.tex",
      settings: {
        foreground: palette.flamingo
      }
    }
  ];
};
var latex_default = tokens14;

// ../catppuccin-vsc/src/theme/tokens/liquid.ts
var tokens15 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Liquid Builtin Objects & User Defined Variables",
      scope: "variable.language.liquid",
      settings: {
        foreground: palette.pink
      }
    }
  ];
};
var liquid_default = tokens15;

// ../catppuccin-vsc/src/theme/tokens/lua.ts
var tokens16 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Lua docstring keywords",
      scope: "comment.line.double-dash.documentation.lua storage.type.annotation.lua",
      settings: {
        foreground: palette.mauve,
        fontStyle: ""
      }
    },
    {
      name: "Lua docstring variables",
      scope: [
        "comment.line.double-dash.documentation.lua entity.name.variable.lua",
        "comment.line.double-dash.documentation.lua variable.lua"
      ],
      settings: {
        foreground: palette.text
      }
    }
  ];
};
var lua_default = tokens16;

// ../catppuccin-vsc/src/theme/tokens/markdown.ts
var tokens17 = (context) => {
  const { palette } = context;
  return [
    {
      scope: [
        "heading.1.markdown punctuation.definition.heading.markdown",
        "heading.1.markdown",
        "heading.1.quarto punctuation.definition.heading.quarto",
        "heading.1.quarto",
        "markup.heading.atx.1.mdx",
        "markup.heading.atx.1.mdx punctuation.definition.heading.mdx",
        "markup.heading.setext.1.markdown",
        "markup.heading.heading-0.asciidoc"
      ],
      settings: {
        foreground: palette.red
      }
    },
    {
      scope: [
        "heading.2.markdown punctuation.definition.heading.markdown",
        "heading.2.markdown",
        "heading.2.quarto punctuation.definition.heading.quarto",
        "heading.2.quarto",
        "markup.heading.atx.2.mdx",
        "markup.heading.atx.2.mdx punctuation.definition.heading.mdx",
        "markup.heading.setext.2.markdown",
        "markup.heading.heading-1.asciidoc"
      ],
      settings: {
        foreground: palette.peach
      }
    },
    {
      scope: [
        "heading.3.markdown punctuation.definition.heading.markdown",
        "heading.3.markdown",
        "heading.3.quarto punctuation.definition.heading.quarto",
        "heading.3.quarto",
        "markup.heading.atx.3.mdx",
        "markup.heading.atx.3.mdx punctuation.definition.heading.mdx",
        "markup.heading.heading-2.asciidoc"
      ],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      scope: [
        "heading.4.markdown punctuation.definition.heading.markdown",
        "heading.4.markdown",
        "heading.4.quarto punctuation.definition.heading.quarto",
        "heading.4.quarto",
        "markup.heading.atx.4.mdx",
        "markup.heading.atx.4.mdx punctuation.definition.heading.mdx",
        "markup.heading.heading-3.asciidoc"
      ],
      settings: {
        foreground: palette.green
      }
    },
    {
      scope: [
        "heading.5.markdown punctuation.definition.heading.markdown",
        "heading.5.markdown",
        "heading.5.quarto punctuation.definition.heading.quarto",
        "heading.5.quarto",
        "markup.heading.atx.5.mdx",
        "markup.heading.atx.5.mdx punctuation.definition.heading.mdx",
        "markup.heading.heading-4.asciidoc"
      ],
      settings: {
        foreground: palette.sapphire
      }
    },
    {
      scope: [
        "heading.6.markdown punctuation.definition.heading.markdown",
        "heading.6.markdown",
        "heading.6.quarto punctuation.definition.heading.quarto",
        "heading.6.quarto",
        "markup.heading.atx.6.mdx",
        "markup.heading.atx.6.mdx punctuation.definition.heading.mdx",
        "markup.heading.heading-5.asciidoc"
      ],
      settings: {
        foreground: palette.lavender
      }
    },
    {
      scope: "markup.bold",
      settings: {
        foreground: palette.red,
        fontStyle: "bold"
      }
    },
    {
      scope: "markup.italic",
      settings: {
        foreground: palette.red,
        fontStyle: "italic"
      }
    },
    {
      scope: "markup.strikethrough",
      settings: {
        foreground: palette.subtext0,
        fontStyle: "strikethrough"
      }
    },
    {
      name: "Markdown auto links",
      scope: ["punctuation.definition.link", "markup.underline.link"],
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "Markdown links",
      scope: [
        "text.html.markdown punctuation.definition.link.title",
        "text.html.quarto punctuation.definition.link.title",
        "string.other.link.title.markdown",
        "string.other.link.title.quarto",
        "markup.link",
        // references like
        // > [1]: http://example.com "Example"
        "punctuation.definition.constant.markdown",
        "punctuation.definition.constant.quarto",
        "constant.other.reference.link.markdown",
        "constant.other.reference.link.quarto",
        "markup.substitution.attribute-reference"
      ],
      settings: {
        foreground: palette.lavender
      }
    },
    {
      name: "Markdown code spans",
      scope: [
        "punctuation.definition.raw.markdown",
        "punctuation.definition.raw.quarto",
        "markup.inline.raw.string.markdown",
        "markup.inline.raw.string.quarto",
        "markup.raw.block.markdown",
        "markup.raw.block.quarto"
      ],
      settings: {
        foreground: palette.green
      }
    },
    {
      name: "Markdown triple backtick language identifier",
      scope: "fenced_code.block.language",
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "Markdown triple backticks",
      scope: [
        "markup.fenced_code.block punctuation.definition",
        "markup.raw support.asciidoc"
      ],
      settings: {
        foreground: palette.overlay2
      }
    },
    {
      name: "Markdown quotes",
      scope: ["markup.quote", "punctuation.definition.quote.begin"],
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "Markdown separators",
      scope: "meta.separator.markdown",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Markdown list bullets",
      scope: [
        "punctuation.definition.list.begin.markdown",
        "punctuation.definition.list.begin.quarto",
        "markup.list.bullet"
      ],
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Quarto headings",
      scope: "markup.heading.quarto",
      settings: {
        fontStyle: "bold"
      }
    }
  ];
};
var markdown_default = tokens17;

// ../catppuccin-vsc/src/theme/tokens/nix.ts
var tokens18 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Nix attribute names",
      scope: [
        "entity.other.attribute-name.multipart.nix",
        "entity.other.attribute-name.single.nix"
      ],
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "Nix parameter names",
      scope: "variable.parameter.name.nix",
      settings: {
        foreground: palette.text,
        fontStyle: ""
      }
    },
    {
      name: "Nix interpolated parameter names",
      scope: "meta.embedded variable.parameter.name.nix",
      settings: {
        foreground: palette.lavender,
        fontStyle: ""
      }
    },
    {
      name: "Nix paths",
      scope: "string.unquoted.path.nix",
      settings: {
        foreground: palette.pink,
        fontStyle: ""
      }
    }
  ];
};
var nix_default = tokens18;

// ../catppuccin-vsc/src/theme/tokens/php.ts
var tokens19 = (context) => {
  const { palette } = context;
  return [
    {
      name: "PHP Attributes",
      scope: ["support.attribute.builtin", "meta.attribute.php"],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "PHP Parameters (needed for the leading dollar sign)",
      scope: "meta.function.parameters.php punctuation.definition.variable.php",
      settings: {
        foreground: palette.maroon
      }
    },
    {
      name: "PHP Constants (null, __FILE__, etc.)",
      scope: "constant.language.php",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "PHP functions",
      scope: "text.html.php support.function",
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "PHPdoc keywords",
      scope: "keyword.other.phpdoc.php",
      settings: {
        fontStyle: ""
      }
    }
  ];
};
var php_default = tokens19;

// ../catppuccin-vsc/src/theme/tokens/python.ts
var tokens20 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Python argument functions reset to text, otherwise they inherit blue from function-call",
      scope: [
        "support.variable.magic.python",
        "meta.function-call.arguments.python"
      ],
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Python double underscore functions",
      scope: ["support.function.magic.python"],
      settings: {
        foreground: palette.sky,
        fontStyle: "italic"
      }
    },
    {
      name: "Python `self` keyword",
      scope: [
        "variable.parameter.function.language.special.self.python",
        "variable.language.special.self.python"
      ],
      settings: {
        foreground: palette.red,
        fontStyle: "italic"
      }
    },
    {
      name: "python keyword flow/logical (for ... in)",
      scope: ["keyword.control.flow.python", "keyword.operator.logical.python"],
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "python storage type",
      scope: "storage.type.function.python",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "python function support",
      scope: [
        "support.token.decorator.python",
        "meta.function.decorator.identifier.python"
      ],
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "python function calls",
      scope: ["meta.function-call.python"],
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "python function decorators",
      scope: [
        "entity.name.function.decorator.python",
        "punctuation.definition.decorator.python"
      ],
      settings: {
        foreground: palette.peach,
        fontStyle: "italic"
      }
    },
    {
      name: "python placeholder reset to normal string",
      scope: "constant.character.format.placeholder.other.python",
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "Python exception & builtins such as exit()",
      scope: [
        "support.type.exception.python",
        "support.function.builtin.python"
      ],
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "entity.name.type",
      scope: ["support.type.python"],
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "python constants (True/False)",
      scope: "constant.language.python",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Arguments accessed later in the function body",
      scope: ["meta.indexed-name.python", "meta.item-access.python"],
      settings: {
        foreground: palette.maroon,
        fontStyle: "italic"
      }
    },
    {
      name: "Python f-strings/binary/unicode storage types",
      scope: "storage.type.string.python",
      settings: {
        foreground: palette.green,
        fontStyle: "italic"
      }
    },
    {
      name: "Python type hints",
      scope: "meta.function.parameters.python",
      settings: {
        fontStyle: ""
      }
    }
  ];
};
var python_default = tokens20;

// ../catppuccin-vsc/src/theme/tokens/r.ts
var tokens21 = (context) => {
  const { palette } = context;
  return [
    {
      name: "R function calls",
      scope: "meta.function-call.r",
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "R function call arguments",
      scope: "meta.function-call.arguments.r",
      settings: {
        foreground: palette.text
      }
    }
  ];
};
var r_default = tokens21;

// ../catppuccin-vsc/src/theme/tokens/regex.ts
var tokens22 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Regex string begin/end in JS/TS",
      scope: [
        "string.regexp punctuation.definition.string.begin",
        "string.regexp punctuation.definition.string.end"
      ],
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "Regex anchors (^, $)",
      scope: "keyword.control.anchor.regexp",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "Regex regular string match",
      scope: "string.regexp.ts",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Regex group parenthesis & backreference (\\1, \\2, \\3, ...)",
      scope: [
        "punctuation.definition.group.regexp",
        "keyword.other.back-reference.regexp"
      ],
      settings: {
        foreground: palette.green
      }
    },
    {
      name: "Regex character class []",
      scope: "punctuation.definition.character-class.regexp",
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Regex character classes (\\d, \\w, \\s)",
      scope: "constant.other.character-class.regexp",
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "Regex range",
      scope: "constant.other.character-class.range.regexp",
      settings: {
        foreground: palette.rosewater
      }
    },
    {
      name: "Regex quantifier",
      scope: "keyword.operator.quantifier.regexp",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Regex constant/numeric",
      scope: "constant.character.numeric.regexp",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Regex lookaheads, negative lookaheads, lookbehinds, negative lookbehinds",
      scope: [
        "punctuation.definition.group.no-capture.regexp",
        "meta.assertion.look-ahead.regexp",
        "meta.assertion.negative-look-ahead.regexp"
      ],
      settings: {
        foreground: palette.blue
      }
    }
  ];
};
var regex_default = tokens22;

// ../catppuccin-vsc/src/theme/tokens/rust.ts
var tokens23 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Rust attribute",
      scope: [
        "meta.annotation.rust",
        "meta.annotation.rust punctuation",
        "meta.attribute.rust",
        "punctuation.definition.attribute.rust"
      ],
      settings: {
        foreground: palette.yellow,
        fontStyle: "italic"
      }
    },
    {
      name: "Rust attribute strings",
      scope: [
        "meta.attribute.rust string.quoted.double.rust",
        "meta.attribute.rust string.quoted.single.char.rust"
      ],
      settings: {
        fontStyle: ""
      }
    },
    {
      name: "Rust keyword",
      scope: [
        "entity.name.function.macro.rules.rust",
        "storage.type.module.rust",
        "storage.modifier.rust",
        "storage.type.struct.rust",
        "storage.type.enum.rust",
        "storage.type.trait.rust",
        "storage.type.union.rust",
        "storage.type.impl.rust",
        "storage.type.rust",
        "storage.type.function.rust",
        "storage.type.type.rust"
      ],
      settings: {
        foreground: palette.mauve,
        fontStyle: ""
      }
    },
    {
      name: "Rust u/i32, u/i64, etc.",
      scope: "entity.name.type.numeric.rust",
      settings: {
        foreground: palette.mauve,
        fontStyle: ""
      }
    },
    {
      name: "Rust generic",
      scope: "meta.generic.rust",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Rust impl",
      scope: "entity.name.impl.rust",
      settings: {
        foreground: palette.yellow,
        fontStyle: "italic"
      }
    },
    {
      name: "Rust module",
      scope: "entity.name.module.rust",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Rust trait",
      scope: "entity.name.trait.rust",
      settings: {
        foreground: palette.yellow,
        fontStyle: "italic"
      }
    },
    {
      name: "Rust struct",
      scope: "storage.type.source.rust",
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Rust union",
      scope: "entity.name.union.rust",
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Rust enum member",
      scope: "meta.enum.rust storage.type.source.rust",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Rust macro",
      scope: [
        "support.macro.rust",
        "meta.macro.rust support.function.rust",
        "entity.name.function.macro.rust"
      ],
      settings: {
        foreground: palette.blue,
        fontStyle: "italic"
      }
    },
    {
      name: "Rust lifetime",
      scope: ["storage.modifier.lifetime.rust", "entity.name.type.lifetime"],
      settings: {
        foreground: palette.blue,
        fontStyle: "italic"
      }
    },
    {
      name: "Rust string formatting",
      scope: "string.quoted.double.rust constant.other.placeholder.rust",
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "Rust return type generic",
      scope: "meta.function.return-type.rust meta.generic.rust storage.type.rust",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Rust functions",
      scope: "meta.function.call.rust",
      settings: {
        foreground: palette.blue
      }
    },
    {
      name: "Rust angle brackets",
      scope: "punctuation.brackets.angle.rust",
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "Rust constants",
      scope: "constant.other.caps.rust",
      settings: {
        foreground: palette.peach
      }
    },
    {
      name: "Rust function parameters",
      scope: ["meta.function.definition.rust variable.other.rust"],
      settings: {
        foreground: palette.maroon
      }
    },
    {
      name: "Rust closure variables",
      scope: "meta.function.call.rust variable.other.rust",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Rust self",
      scope: "variable.language.self.rust",
      settings: {
        foreground: palette.red
      }
    },
    {
      name: "Rust metavariable names",
      scope: [
        "variable.other.metavariable.name.rust",
        "meta.macro.metavariable.rust keyword.operator.macro.dollar.rust"
      ],
      settings: {
        foreground: palette.pink
      }
    }
  ];
};
var rust_default = tokens23;

// ../catppuccin-vsc/src/theme/tokens/shell.ts
var tokens24 = (context) => {
  const { palette } = context;
  return [
    {
      name: "Shell shebang",
      scope: [
        "comment.line.shebang",
        "comment.line.shebang punctuation.definition.comment",
        "comment.line.shebang",
        "punctuation.definition.comment.shebang.shell",
        "meta.shebang.shell"
      ],
      settings: {
        foreground: palette.pink,
        fontStyle: "italic"
      }
    },
    {
      name: "Shell shebang command",
      scope: "comment.line.shebang constant.language",
      settings: {
        foreground: palette.teal,
        fontStyle: "italic"
      }
    },
    {
      name: "Shell interpolated command",
      scope: [
        "meta.function-call.arguments.shell punctuation.definition.variable.shell",
        "meta.function-call.arguments.shell punctuation.section.interpolation",
        "meta.function-call.arguments.shell punctuation.definition.variable.shell",
        "meta.function-call.arguments.shell punctuation.section.interpolation"
      ],
      settings: {
        foreground: palette.red
      }
    },
    {
      name: "Shell interpolated command variable",
      scope: "meta.string meta.interpolation.parameter.shell variable.other.readwrite",
      settings: {
        foreground: palette.peach,
        fontStyle: "italic"
      }
    },
    {
      scope: [
        "source.shell punctuation.section.interpolation",
        "punctuation.definition.evaluation.backticks.shell"
      ],
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Shell EOF",
      scope: "entity.name.tag.heredoc.shell",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "Shell quoted variable",
      scope: "string.quoted.double.shell variable.other.normal.shell",
      settings: {
        foreground: palette.text
      }
    }
  ];
};
var shell_default = tokens24;

// ../catppuccin-vsc/src/theme/tokens/typst.ts
var tokens25 = (context) => {
  const { palette } = context;
  return [
    {
      scope: ["markup.heading.typst"],
      settings: {
        foreground: palette.red
      }
    }
  ];
};
var typst_default = tokens25;

// ../catppuccin-vsc/src/theme/tokens/index.ts
function tokens26(context) {
  const { options, palette } = context;
  return [
    {
      name: "Basic text & variable names (incl. leading punctuation)",
      scope: [
        "text",
        "source",
        "variable.other.readwrite",
        "punctuation.definition.variable"
      ],
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "Parentheses, Brackets, Braces",
      scope: "punctuation",
      settings: {
        foreground: palette.overlay2,
        fontStyle: ""
      }
    },
    {
      name: "Comments",
      scope: ["comment", "punctuation.definition.comment"],
      settings: {
        foreground: palette.overlay2,
        fontStyle: options.italicComments ? "italic" : ""
      }
    },
    {
      scope: ["string", "punctuation.definition.string"],
      settings: {
        foreground: palette.green
      }
    },
    {
      scope: "constant.character.escape",
      settings: {
        foreground: palette.pink
      }
    },
    {
      name: "Booleans, constants, numbers",
      scope: [
        "constant.numeric",
        "variable.other.constant",
        "entity.name.constant",
        "constant.language.boolean",
        "constant.language.false",
        "constant.language.true",
        "keyword.other.unit.user-defined",
        "keyword.other.unit.suffix.floating-point"
      ],
      settings: {
        foreground: palette.peach
      }
    },
    {
      scope: [
        "keyword",
        "keyword.operator.word",
        "keyword.operator.new",
        "variable.language.super",
        "support.type.primitive",
        "storage.type",
        "storage.modifier",
        // include punctuation like $ and @ if they're part of the keyword
        "punctuation.definition.keyword"
      ],
      settings: {
        foreground: palette.mauve,
        fontStyle: ""
      }
    },
    {
      scope: "entity.name.tag.documentation",
      settings: {
        foreground: palette.mauve
      }
    },
    {
      name: "Punctuation",
      scope: [
        "keyword.operator",
        "punctuation.accessor",
        "punctuation.definition.generic",
        "meta.function.closure punctuation.section.parameters",
        "punctuation.definition.tag",
        "punctuation.separator.key-value"
      ],
      settings: {
        foreground: palette.teal
      }
    },
    {
      scope: [
        "entity.name.function",
        "meta.function-call.method",
        "support.function",
        "support.function.misc",
        "variable.function"
      ],
      settings: {
        foreground: palette.blue,
        fontStyle: "italic"
      }
    },
    {
      name: "Classes",
      scope: [
        "entity.name.class",
        "entity.other.inherited-class",
        "support.class",
        "meta.function-call.constructor",
        "entity.name.struct"
      ],
      settings: {
        foreground: palette.yellow,
        fontStyle: "italic"
      }
    },
    {
      name: "Enum",
      scope: "entity.name.enum",
      settings: {
        foreground: palette.yellow,
        fontStyle: "italic"
      }
    },
    {
      name: "Enum member",
      scope: [
        "meta.enum variable.other.readwrite",
        "variable.other.enummember"
      ],
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Object properties",
      scope: "meta.property.object",
      settings: {
        foreground: palette.teal
      }
    },
    {
      name: "Types",
      scope: [
        "meta.type",
        "meta.type-alias",
        "support.type",
        "entity.name.type"
      ],
      settings: {
        foreground: palette.yellow,
        fontStyle: "italic"
      }
    },
    {
      name: "Decorators",
      scope: [
        "meta.annotation variable.function",
        "meta.annotation variable.annotation.function",
        "meta.annotation punctuation.definition.annotation",
        "meta.decorator",
        "punctuation.decorator"
      ],
      settings: {
        foreground: palette.peach
      }
    },
    {
      scope: ["variable.parameter", "meta.function.parameters"],
      settings: {
        foreground: palette.maroon,
        fontStyle: "italic"
      }
    },
    {
      name: "Built-ins",
      scope: ["constant.language", "support.function.builtin"],
      settings: {
        foreground: palette.red
      }
    },
    {
      scope: "entity.other.attribute-name.documentation",
      settings: {
        foreground: palette.red
      }
    },
    {
      name: "Preprocessor directives",
      scope: ["keyword.control.directive", "punctuation.definition.directive"],
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Type parameters",
      scope: "punctuation.definition.typeparameters",
      settings: {
        foreground: palette.sky
      }
    },
    {
      name: "Namespaces",
      scope: "entity.name.namespace",
      settings: {
        foreground: palette.yellow
      }
    },
    {
      name: "Property names (left hand assignments in json/yaml/css/less)",
      scope: [
        "support.type.property-name.css",
        "support.type.property-name.less"
      ],
      settings: {
        foreground: palette.blue,
        fontStyle: ""
      }
    },
    {
      name: "This/Self keyword",
      scope: [
        "variable.language.this",
        // leading punctuation like $this in PHP
        "variable.language.this punctuation.definition.variable"
      ],
      settings: {
        foreground: palette.red
      }
    },
    {
      name: "Object properties",
      scope: "variable.object.property",
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "String template interpolation",
      scope: ["string.template variable", "string variable"],
      settings: {
        foreground: palette.text
      }
    },
    {
      name: "`new` as bold",
      scope: "keyword.operator.new",
      settings: {
        fontStyle: "bold"
      }
    },
    // per-language tokens
    ...[
      cpp_default,
      cs_default,
      css_default,
      data_default,
      diff_default,
      dotenv_default,
      gdscript_default,
      golang_default,
      graphql_default,
      html_default,
      java_default,
      javascript_default,
      julia_default,
      latex_default,
      liquid_default,
      lua_default,
      markdown_default,
      nix_default,
      php_default,
      python_default,
      r_default,
      regex_default,
      rust_default,
      shell_default,
      typst_default
    ].flatMap((element) => element(context))
  ];
}

// ../catppuccin-vsc/src/theme/tokenColors.ts
var getTokenColors = (context) => {
  const { options } = context;
  return tokens26(context).map((token) => {
    if (token.name === "Comments") return token;
    let fontStyle = token.settings.fontStyle;
    if (!options.italicKeywords && fontStyle !== void 0) {
      fontStyle = fontStyle == null ? void 0 : fontStyle.replace("italic", "");
    }
    if (!options.boldKeywords && fontStyle !== void 0 && token.scope !== "markup.bold") {
      fontStyle = fontStyle == null ? void 0 : fontStyle.replace("bold", "");
    }
    token = {
      ...token,
      settings: { ...token.settings, fontStyle }
    };
    return token;
  });
};

// ../catppuccin-vsc/src/theme/semanticTokens.ts
var getSemanticTokens = (context) => {
  const { palette } = context;
  return {
    enumMember: { foreground: palette.teal },
    selfKeyword: { foreground: palette.red },
    boolean: { foreground: palette.peach },
    number: { foreground: palette.peach },
    "variable.defaultLibrary": { foreground: palette.maroon },
    // Python types
    "class:python": { foreground: palette.yellow },
    "class.builtin:python": { foreground: palette.mauve },
    "variable.typeHint:python": { foreground: palette.yellow },
    "function.decorator:python": { foreground: palette.peach },
    // ignore `const`s being peach in JS & TS
    "variable.readonly:javascript": { foreground: palette.text },
    "variable.readonly:typescript": { foreground: palette.text },
    "property.readonly:javascript": { foreground: palette.text },
    "property.readonly:typescript": { foreground: palette.text },
    // ditto for React
    "variable.readonly:javascriptreact": { foreground: palette.text },
    "variable.readonly:typescriptreact": { foreground: palette.text },
    "property.readonly:javascriptreact": { foreground: palette.text },
    "property.readonly:typescriptreact": { foreground: palette.text },
    // Scala, also dealing with constants
    "variable.readonly:scala": { foreground: palette.text },
    // Golang builtin readonly defaultLibrary (nil)
    "type.defaultLibrary:go": { foreground: palette.mauve },
    "variable.readonly.defaultLibrary:go": { foreground: palette.mauve },
    // TOML syntax
    tomlArrayKey: { foreground: palette.blue, fontStyle: "" },
    tomlTableKey: { foreground: palette.blue, fontStyle: "" },
    // Rust attributes
    "builtinAttribute.attribute.library:rust": { foreground: palette.blue },
    "generic.attribute:rust": { foreground: palette.text },
    // Nix
    "constant.builtin.readonly:nix": { foreground: palette.mauve },
    // Typst
    heading: { foreground: palette.red },
    "text.emph": { foreground: palette.red, fontStyle: "italic" },
    "text.strong": { foreground: palette.red, fontStyle: "bold" },
    "text.math": { foreground: palette.flamingo },
    pol: { foreground: palette.flamingo }
  };
};

// ../../node_modules/.pnpm/tinycolor2@1.6.0/node_modules/tinycolor2/esm/tinycolor.js
function _typeof(obj) {
  "@babel/helpers - typeof";
  return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(obj2) {
    return typeof obj2;
  } : function(obj2) {
    return obj2 && "function" == typeof Symbol && obj2.constructor === Symbol && obj2 !== Symbol.prototype ? "symbol" : typeof obj2;
  }, _typeof(obj);
}
var trimLeft = /^\s+/;
var trimRight = /\s+$/;
function tinycolor(color, opts) {
  color = color ? color : "";
  opts = opts || {};
  if (color instanceof tinycolor) {
    return color;
  }
  if (!(this instanceof tinycolor)) {
    return new tinycolor(color, opts);
  }
  var rgb = inputToRGB(color);
  this._originalInput = color, this._r = rgb.r, this._g = rgb.g, this._b = rgb.b, this._a = rgb.a, this._roundA = Math.round(100 * this._a) / 100, this._format = opts.format || rgb.format;
  this._gradientType = opts.gradientType;
  if (this._r < 1) this._r = Math.round(this._r);
  if (this._g < 1) this._g = Math.round(this._g);
  if (this._b < 1) this._b = Math.round(this._b);
  this._ok = rgb.ok;
}
tinycolor.prototype = {
  isDark: function isDark() {
    return this.getBrightness() < 128;
  },
  isLight: function isLight() {
    return !this.isDark();
  },
  isValid: function isValid() {
    return this._ok;
  },
  getOriginalInput: function getOriginalInput() {
    return this._originalInput;
  },
  getFormat: function getFormat() {
    return this._format;
  },
  getAlpha: function getAlpha() {
    return this._a;
  },
  getBrightness: function getBrightness() {
    var rgb = this.toRgb();
    return (rgb.r * 299 + rgb.g * 587 + rgb.b * 114) / 1e3;
  },
  getLuminance: function getLuminance() {
    var rgb = this.toRgb();
    var RsRGB, GsRGB, BsRGB, R, G, B;
    RsRGB = rgb.r / 255;
    GsRGB = rgb.g / 255;
    BsRGB = rgb.b / 255;
    if (RsRGB <= 0.03928) R = RsRGB / 12.92;
    else R = Math.pow((RsRGB + 0.055) / 1.055, 2.4);
    if (GsRGB <= 0.03928) G = GsRGB / 12.92;
    else G = Math.pow((GsRGB + 0.055) / 1.055, 2.4);
    if (BsRGB <= 0.03928) B = BsRGB / 12.92;
    else B = Math.pow((BsRGB + 0.055) / 1.055, 2.4);
    return 0.2126 * R + 0.7152 * G + 0.0722 * B;
  },
  setAlpha: function setAlpha(value) {
    this._a = boundAlpha(value);
    this._roundA = Math.round(100 * this._a) / 100;
    return this;
  },
  toHsv: function toHsv() {
    var hsv = rgbToHsv(this._r, this._g, this._b);
    return {
      h: hsv.h * 360,
      s: hsv.s,
      v: hsv.v,
      a: this._a
    };
  },
  toHsvString: function toHsvString() {
    var hsv = rgbToHsv(this._r, this._g, this._b);
    var h = Math.round(hsv.h * 360), s = Math.round(hsv.s * 100), v = Math.round(hsv.v * 100);
    return this._a == 1 ? "hsv(" + h + ", " + s + "%, " + v + "%)" : "hsva(" + h + ", " + s + "%, " + v + "%, " + this._roundA + ")";
  },
  toHsl: function toHsl() {
    var hsl = rgbToHsl(this._r, this._g, this._b);
    return {
      h: hsl.h * 360,
      s: hsl.s,
      l: hsl.l,
      a: this._a
    };
  },
  toHslString: function toHslString() {
    var hsl = rgbToHsl(this._r, this._g, this._b);
    var h = Math.round(hsl.h * 360), s = Math.round(hsl.s * 100), l = Math.round(hsl.l * 100);
    return this._a == 1 ? "hsl(" + h + ", " + s + "%, " + l + "%)" : "hsla(" + h + ", " + s + "%, " + l + "%, " + this._roundA + ")";
  },
  toHex: function toHex(allow3Char) {
    return rgbToHex(this._r, this._g, this._b, allow3Char);
  },
  toHexString: function toHexString(allow3Char) {
    return "#" + this.toHex(allow3Char);
  },
  toHex8: function toHex8(allow4Char) {
    return rgbaToHex(this._r, this._g, this._b, this._a, allow4Char);
  },
  toHex8String: function toHex8String(allow4Char) {
    return "#" + this.toHex8(allow4Char);
  },
  toRgb: function toRgb() {
    return {
      r: Math.round(this._r),
      g: Math.round(this._g),
      b: Math.round(this._b),
      a: this._a
    };
  },
  toRgbString: function toRgbString() {
    return this._a == 1 ? "rgb(" + Math.round(this._r) + ", " + Math.round(this._g) + ", " + Math.round(this._b) + ")" : "rgba(" + Math.round(this._r) + ", " + Math.round(this._g) + ", " + Math.round(this._b) + ", " + this._roundA + ")";
  },
  toPercentageRgb: function toPercentageRgb() {
    return {
      r: Math.round(bound01(this._r, 255) * 100) + "%",
      g: Math.round(bound01(this._g, 255) * 100) + "%",
      b: Math.round(bound01(this._b, 255) * 100) + "%",
      a: this._a
    };
  },
  toPercentageRgbString: function toPercentageRgbString() {
    return this._a == 1 ? "rgb(" + Math.round(bound01(this._r, 255) * 100) + "%, " + Math.round(bound01(this._g, 255) * 100) + "%, " + Math.round(bound01(this._b, 255) * 100) + "%)" : "rgba(" + Math.round(bound01(this._r, 255) * 100) + "%, " + Math.round(bound01(this._g, 255) * 100) + "%, " + Math.round(bound01(this._b, 255) * 100) + "%, " + this._roundA + ")";
  },
  toName: function toName() {
    if (this._a === 0) {
      return "transparent";
    }
    if (this._a < 1) {
      return false;
    }
    return hexNames[rgbToHex(this._r, this._g, this._b, true)] || false;
  },
  toFilter: function toFilter(secondColor) {
    var hex8String = "#" + rgbaToArgbHex(this._r, this._g, this._b, this._a);
    var secondHex8String = hex8String;
    var gradientType = this._gradientType ? "GradientType = 1, " : "";
    if (secondColor) {
      var s = tinycolor(secondColor);
      secondHex8String = "#" + rgbaToArgbHex(s._r, s._g, s._b, s._a);
    }
    return "progid:DXImageTransform.Microsoft.gradient(" + gradientType + "startColorstr=" + hex8String + ",endColorstr=" + secondHex8String + ")";
  },
  toString: function toString(format) {
    var formatSet = !!format;
    format = format || this._format;
    var formattedString = false;
    var hasAlpha = this._a < 1 && this._a >= 0;
    var needsAlphaFormat = !formatSet && hasAlpha && (format === "hex" || format === "hex6" || format === "hex3" || format === "hex4" || format === "hex8" || format === "name");
    if (needsAlphaFormat) {
      if (format === "name" && this._a === 0) {
        return this.toName();
      }
      return this.toRgbString();
    }
    if (format === "rgb") {
      formattedString = this.toRgbString();
    }
    if (format === "prgb") {
      formattedString = this.toPercentageRgbString();
    }
    if (format === "hex" || format === "hex6") {
      formattedString = this.toHexString();
    }
    if (format === "hex3") {
      formattedString = this.toHexString(true);
    }
    if (format === "hex4") {
      formattedString = this.toHex8String(true);
    }
    if (format === "hex8") {
      formattedString = this.toHex8String();
    }
    if (format === "name") {
      formattedString = this.toName();
    }
    if (format === "hsl") {
      formattedString = this.toHslString();
    }
    if (format === "hsv") {
      formattedString = this.toHsvString();
    }
    return formattedString || this.toHexString();
  },
  clone: function clone() {
    return tinycolor(this.toString());
  },
  _applyModification: function _applyModification(fn, args) {
    var color = fn.apply(null, [this].concat([].slice.call(args)));
    this._r = color._r;
    this._g = color._g;
    this._b = color._b;
    this.setAlpha(color._a);
    return this;
  },
  lighten: function lighten() {
    return this._applyModification(_lighten, arguments);
  },
  brighten: function brighten() {
    return this._applyModification(_brighten, arguments);
  },
  darken: function darken() {
    return this._applyModification(_darken, arguments);
  },
  desaturate: function desaturate() {
    return this._applyModification(_desaturate, arguments);
  },
  saturate: function saturate() {
    return this._applyModification(_saturate, arguments);
  },
  greyscale: function greyscale() {
    return this._applyModification(_greyscale, arguments);
  },
  spin: function spin() {
    return this._applyModification(_spin, arguments);
  },
  _applyCombination: function _applyCombination(fn, args) {
    return fn.apply(null, [this].concat([].slice.call(args)));
  },
  analogous: function analogous() {
    return this._applyCombination(_analogous, arguments);
  },
  complement: function complement() {
    return this._applyCombination(_complement, arguments);
  },
  monochromatic: function monochromatic() {
    return this._applyCombination(_monochromatic, arguments);
  },
  splitcomplement: function splitcomplement() {
    return this._applyCombination(_splitcomplement, arguments);
  },
  // Disabled until https://github.com/bgrins/TinyColor/issues/254
  // polyad: function (number) {
  //   return this._applyCombination(polyad, [number]);
  // },
  triad: function triad() {
    return this._applyCombination(polyad, [3]);
  },
  tetrad: function tetrad() {
    return this._applyCombination(polyad, [4]);
  }
};
tinycolor.fromRatio = function(color, opts) {
  if (_typeof(color) == "object") {
    var newColor = {};
    for (var i in color) {
      if (color.hasOwnProperty(i)) {
        if (i === "a") {
          newColor[i] = color[i];
        } else {
          newColor[i] = convertToPercentage(color[i]);
        }
      }
    }
    color = newColor;
  }
  return tinycolor(color, opts);
};
function inputToRGB(color) {
  var rgb = {
    r: 0,
    g: 0,
    b: 0
  };
  var a = 1;
  var s = null;
  var v = null;
  var l = null;
  var ok = false;
  var format = false;
  if (typeof color == "string") {
    color = stringInputToObject(color);
  }
  if (_typeof(color) == "object") {
    if (isValidCSSUnit(color.r) && isValidCSSUnit(color.g) && isValidCSSUnit(color.b)) {
      rgb = rgbToRgb(color.r, color.g, color.b);
      ok = true;
      format = String(color.r).substr(-1) === "%" ? "prgb" : "rgb";
    } else if (isValidCSSUnit(color.h) && isValidCSSUnit(color.s) && isValidCSSUnit(color.v)) {
      s = convertToPercentage(color.s);
      v = convertToPercentage(color.v);
      rgb = hsvToRgb(color.h, s, v);
      ok = true;
      format = "hsv";
    } else if (isValidCSSUnit(color.h) && isValidCSSUnit(color.s) && isValidCSSUnit(color.l)) {
      s = convertToPercentage(color.s);
      l = convertToPercentage(color.l);
      rgb = hslToRgb(color.h, s, l);
      ok = true;
      format = "hsl";
    }
    if (color.hasOwnProperty("a")) {
      a = color.a;
    }
  }
  a = boundAlpha(a);
  return {
    ok,
    format: color.format || format,
    r: Math.min(255, Math.max(rgb.r, 0)),
    g: Math.min(255, Math.max(rgb.g, 0)),
    b: Math.min(255, Math.max(rgb.b, 0)),
    a
  };
}
function rgbToRgb(r, g, b) {
  return {
    r: bound01(r, 255) * 255,
    g: bound01(g, 255) * 255,
    b: bound01(b, 255) * 255
  };
}
function rgbToHsl(r, g, b) {
  r = bound01(r, 255);
  g = bound01(g, 255);
  b = bound01(b, 255);
  var max = Math.max(r, g, b), min = Math.min(r, g, b);
  var h, s, l = (max + min) / 2;
  if (max == min) {
    h = s = 0;
  } else {
    var d = max - min;
    s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
    switch (max) {
      case r:
        h = (g - b) / d + (g < b ? 6 : 0);
        break;
      case g:
        h = (b - r) / d + 2;
        break;
      case b:
        h = (r - g) / d + 4;
        break;
    }
    h /= 6;
  }
  return {
    h,
    s,
    l
  };
}
function hslToRgb(h, s, l) {
  var r, g, b;
  h = bound01(h, 360);
  s = bound01(s, 100);
  l = bound01(l, 100);
  function hue2rgb(p2, q2, t) {
    if (t < 0) t += 1;
    if (t > 1) t -= 1;
    if (t < 1 / 6) return p2 + (q2 - p2) * 6 * t;
    if (t < 1 / 2) return q2;
    if (t < 2 / 3) return p2 + (q2 - p2) * (2 / 3 - t) * 6;
    return p2;
  }
  if (s === 0) {
    r = g = b = l;
  } else {
    var q = l < 0.5 ? l * (1 + s) : l + s - l * s;
    var p = 2 * l - q;
    r = hue2rgb(p, q, h + 1 / 3);
    g = hue2rgb(p, q, h);
    b = hue2rgb(p, q, h - 1 / 3);
  }
  return {
    r: r * 255,
    g: g * 255,
    b: b * 255
  };
}
function rgbToHsv(r, g, b) {
  r = bound01(r, 255);
  g = bound01(g, 255);
  b = bound01(b, 255);
  var max = Math.max(r, g, b), min = Math.min(r, g, b);
  var h, s, v = max;
  var d = max - min;
  s = max === 0 ? 0 : d / max;
  if (max == min) {
    h = 0;
  } else {
    switch (max) {
      case r:
        h = (g - b) / d + (g < b ? 6 : 0);
        break;
      case g:
        h = (b - r) / d + 2;
        break;
      case b:
        h = (r - g) / d + 4;
        break;
    }
    h /= 6;
  }
  return {
    h,
    s,
    v
  };
}
function hsvToRgb(h, s, v) {
  h = bound01(h, 360) * 6;
  s = bound01(s, 100);
  v = bound01(v, 100);
  var i = Math.floor(h), f = h - i, p = v * (1 - s), q = v * (1 - f * s), t = v * (1 - (1 - f) * s), mod = i % 6, r = [v, q, p, p, t, v][mod], g = [t, v, v, q, p, p][mod], b = [p, p, t, v, v, q][mod];
  return {
    r: r * 255,
    g: g * 255,
    b: b * 255
  };
}
function rgbToHex(r, g, b, allow3Char) {
  var hex = [pad2(Math.round(r).toString(16)), pad2(Math.round(g).toString(16)), pad2(Math.round(b).toString(16))];
  if (allow3Char && hex[0].charAt(0) == hex[0].charAt(1) && hex[1].charAt(0) == hex[1].charAt(1) && hex[2].charAt(0) == hex[2].charAt(1)) {
    return hex[0].charAt(0) + hex[1].charAt(0) + hex[2].charAt(0);
  }
  return hex.join("");
}
function rgbaToHex(r, g, b, a, allow4Char) {
  var hex = [pad2(Math.round(r).toString(16)), pad2(Math.round(g).toString(16)), pad2(Math.round(b).toString(16)), pad2(convertDecimalToHex(a))];
  if (allow4Char && hex[0].charAt(0) == hex[0].charAt(1) && hex[1].charAt(0) == hex[1].charAt(1) && hex[2].charAt(0) == hex[2].charAt(1) && hex[3].charAt(0) == hex[3].charAt(1)) {
    return hex[0].charAt(0) + hex[1].charAt(0) + hex[2].charAt(0) + hex[3].charAt(0);
  }
  return hex.join("");
}
function rgbaToArgbHex(r, g, b, a) {
  var hex = [pad2(convertDecimalToHex(a)), pad2(Math.round(r).toString(16)), pad2(Math.round(g).toString(16)), pad2(Math.round(b).toString(16))];
  return hex.join("");
}
tinycolor.equals = function(color1, color2) {
  if (!color1 || !color2) return false;
  return tinycolor(color1).toRgbString() == tinycolor(color2).toRgbString();
};
tinycolor.random = function() {
  return tinycolor.fromRatio({
    r: Math.random(),
    g: Math.random(),
    b: Math.random()
  });
};
function _desaturate(color, amount) {
  amount = amount === 0 ? 0 : amount || 10;
  var hsl = tinycolor(color).toHsl();
  hsl.s -= amount / 100;
  hsl.s = clamp01(hsl.s);
  return tinycolor(hsl);
}
function _saturate(color, amount) {
  amount = amount === 0 ? 0 : amount || 10;
  var hsl = tinycolor(color).toHsl();
  hsl.s += amount / 100;
  hsl.s = clamp01(hsl.s);
  return tinycolor(hsl);
}
function _greyscale(color) {
  return tinycolor(color).desaturate(100);
}
function _lighten(color, amount) {
  amount = amount === 0 ? 0 : amount || 10;
  var hsl = tinycolor(color).toHsl();
  hsl.l += amount / 100;
  hsl.l = clamp01(hsl.l);
  return tinycolor(hsl);
}
function _brighten(color, amount) {
  amount = amount === 0 ? 0 : amount || 10;
  var rgb = tinycolor(color).toRgb();
  rgb.r = Math.max(0, Math.min(255, rgb.r - Math.round(255 * -(amount / 100))));
  rgb.g = Math.max(0, Math.min(255, rgb.g - Math.round(255 * -(amount / 100))));
  rgb.b = Math.max(0, Math.min(255, rgb.b - Math.round(255 * -(amount / 100))));
  return tinycolor(rgb);
}
function _darken(color, amount) {
  amount = amount === 0 ? 0 : amount || 10;
  var hsl = tinycolor(color).toHsl();
  hsl.l -= amount / 100;
  hsl.l = clamp01(hsl.l);
  return tinycolor(hsl);
}
function _spin(color, amount) {
  var hsl = tinycolor(color).toHsl();
  var hue = (hsl.h + amount) % 360;
  hsl.h = hue < 0 ? 360 + hue : hue;
  return tinycolor(hsl);
}
function _complement(color) {
  var hsl = tinycolor(color).toHsl();
  hsl.h = (hsl.h + 180) % 360;
  return tinycolor(hsl);
}
function polyad(color, number) {
  if (isNaN(number) || number <= 0) {
    throw new Error("Argument to polyad must be a positive number");
  }
  var hsl = tinycolor(color).toHsl();
  var result = [tinycolor(color)];
  var step = 360 / number;
  for (var i = 1; i < number; i++) {
    result.push(tinycolor({
      h: (hsl.h + i * step) % 360,
      s: hsl.s,
      l: hsl.l
    }));
  }
  return result;
}
function _splitcomplement(color) {
  var hsl = tinycolor(color).toHsl();
  var h = hsl.h;
  return [tinycolor(color), tinycolor({
    h: (h + 72) % 360,
    s: hsl.s,
    l: hsl.l
  }), tinycolor({
    h: (h + 216) % 360,
    s: hsl.s,
    l: hsl.l
  })];
}
function _analogous(color, results, slices) {
  results = results || 6;
  slices = slices || 30;
  var hsl = tinycolor(color).toHsl();
  var part = 360 / slices;
  var ret = [tinycolor(color)];
  for (hsl.h = (hsl.h - (part * results >> 1) + 720) % 360; --results; ) {
    hsl.h = (hsl.h + part) % 360;
    ret.push(tinycolor(hsl));
  }
  return ret;
}
function _monochromatic(color, results) {
  results = results || 6;
  var hsv = tinycolor(color).toHsv();
  var h = hsv.h, s = hsv.s, v = hsv.v;
  var ret = [];
  var modification = 1 / results;
  while (results--) {
    ret.push(tinycolor({
      h,
      s,
      v
    }));
    v = (v + modification) % 1;
  }
  return ret;
}
tinycolor.mix = function(color1, color2, amount) {
  amount = amount === 0 ? 0 : amount || 50;
  var rgb1 = tinycolor(color1).toRgb();
  var rgb2 = tinycolor(color2).toRgb();
  var p = amount / 100;
  var rgba = {
    r: (rgb2.r - rgb1.r) * p + rgb1.r,
    g: (rgb2.g - rgb1.g) * p + rgb1.g,
    b: (rgb2.b - rgb1.b) * p + rgb1.b,
    a: (rgb2.a - rgb1.a) * p + rgb1.a
  };
  return tinycolor(rgba);
};
tinycolor.readability = function(color1, color2) {
  var c1 = tinycolor(color1);
  var c2 = tinycolor(color2);
  return (Math.max(c1.getLuminance(), c2.getLuminance()) + 0.05) / (Math.min(c1.getLuminance(), c2.getLuminance()) + 0.05);
};
tinycolor.isReadable = function(color1, color2, wcag2) {
  var readability = tinycolor.readability(color1, color2);
  var wcag2Parms, out;
  out = false;
  wcag2Parms = validateWCAG2Parms(wcag2);
  switch (wcag2Parms.level + wcag2Parms.size) {
    case "AAsmall":
    case "AAAlarge":
      out = readability >= 4.5;
      break;
    case "AAlarge":
      out = readability >= 3;
      break;
    case "AAAsmall":
      out = readability >= 7;
      break;
  }
  return out;
};
tinycolor.mostReadable = function(baseColor, colorList, args) {
  var bestColor = null;
  var bestScore = 0;
  var readability;
  var includeFallbackColors, level, size;
  args = args || {};
  includeFallbackColors = args.includeFallbackColors;
  level = args.level;
  size = args.size;
  for (var i = 0; i < colorList.length; i++) {
    readability = tinycolor.readability(baseColor, colorList[i]);
    if (readability > bestScore) {
      bestScore = readability;
      bestColor = tinycolor(colorList[i]);
    }
  }
  if (tinycolor.isReadable(baseColor, bestColor, {
    level,
    size
  }) || !includeFallbackColors) {
    return bestColor;
  } else {
    args.includeFallbackColors = false;
    return tinycolor.mostReadable(baseColor, ["#fff", "#000"], args);
  }
};
var names = tinycolor.names = {
  aliceblue: "f0f8ff",
  antiquewhite: "faebd7",
  aqua: "0ff",
  aquamarine: "7fffd4",
  azure: "f0ffff",
  beige: "f5f5dc",
  bisque: "ffe4c4",
  black: "000",
  blanchedalmond: "ffebcd",
  blue: "00f",
  blueviolet: "8a2be2",
  brown: "a52a2a",
  burlywood: "deb887",
  burntsienna: "ea7e5d",
  cadetblue: "5f9ea0",
  chartreuse: "7fff00",
  chocolate: "d2691e",
  coral: "ff7f50",
  cornflowerblue: "6495ed",
  cornsilk: "fff8dc",
  crimson: "dc143c",
  cyan: "0ff",
  darkblue: "00008b",
  darkcyan: "008b8b",
  darkgoldenrod: "b8860b",
  darkgray: "a9a9a9",
  darkgreen: "006400",
  darkgrey: "a9a9a9",
  darkkhaki: "bdb76b",
  darkmagenta: "8b008b",
  darkolivegreen: "556b2f",
  darkorange: "ff8c00",
  darkorchid: "9932cc",
  darkred: "8b0000",
  darksalmon: "e9967a",
  darkseagreen: "8fbc8f",
  darkslateblue: "483d8b",
  darkslategray: "2f4f4f",
  darkslategrey: "2f4f4f",
  darkturquoise: "00ced1",
  darkviolet: "9400d3",
  deeppink: "ff1493",
  deepskyblue: "00bfff",
  dimgray: "696969",
  dimgrey: "696969",
  dodgerblue: "1e90ff",
  firebrick: "b22222",
  floralwhite: "fffaf0",
  forestgreen: "228b22",
  fuchsia: "f0f",
  gainsboro: "dcdcdc",
  ghostwhite: "f8f8ff",
  gold: "ffd700",
  goldenrod: "daa520",
  gray: "808080",
  green: "008000",
  greenyellow: "adff2f",
  grey: "808080",
  honeydew: "f0fff0",
  hotpink: "ff69b4",
  indianred: "cd5c5c",
  indigo: "4b0082",
  ivory: "fffff0",
  khaki: "f0e68c",
  lavender: "e6e6fa",
  lavenderblush: "fff0f5",
  lawngreen: "7cfc00",
  lemonchiffon: "fffacd",
  lightblue: "add8e6",
  lightcoral: "f08080",
  lightcyan: "e0ffff",
  lightgoldenrodyellow: "fafad2",
  lightgray: "d3d3d3",
  lightgreen: "90ee90",
  lightgrey: "d3d3d3",
  lightpink: "ffb6c1",
  lightsalmon: "ffa07a",
  lightseagreen: "20b2aa",
  lightskyblue: "87cefa",
  lightslategray: "789",
  lightslategrey: "789",
  lightsteelblue: "b0c4de",
  lightyellow: "ffffe0",
  lime: "0f0",
  limegreen: "32cd32",
  linen: "faf0e6",
  magenta: "f0f",
  maroon: "800000",
  mediumaquamarine: "66cdaa",
  mediumblue: "0000cd",
  mediumorchid: "ba55d3",
  mediumpurple: "9370db",
  mediumseagreen: "3cb371",
  mediumslateblue: "7b68ee",
  mediumspringgreen: "00fa9a",
  mediumturquoise: "48d1cc",
  mediumvioletred: "c71585",
  midnightblue: "191970",
  mintcream: "f5fffa",
  mistyrose: "ffe4e1",
  moccasin: "ffe4b5",
  navajowhite: "ffdead",
  navy: "000080",
  oldlace: "fdf5e6",
  olive: "808000",
  olivedrab: "6b8e23",
  orange: "ffa500",
  orangered: "ff4500",
  orchid: "da70d6",
  palegoldenrod: "eee8aa",
  palegreen: "98fb98",
  paleturquoise: "afeeee",
  palevioletred: "db7093",
  papayawhip: "ffefd5",
  peachpuff: "ffdab9",
  peru: "cd853f",
  pink: "ffc0cb",
  plum: "dda0dd",
  powderblue: "b0e0e6",
  purple: "800080",
  rebeccapurple: "663399",
  red: "f00",
  rosybrown: "bc8f8f",
  royalblue: "4169e1",
  saddlebrown: "8b4513",
  salmon: "fa8072",
  sandybrown: "f4a460",
  seagreen: "2e8b57",
  seashell: "fff5ee",
  sienna: "a0522d",
  silver: "c0c0c0",
  skyblue: "87ceeb",
  slateblue: "6a5acd",
  slategray: "708090",
  slategrey: "708090",
  snow: "fffafa",
  springgreen: "00ff7f",
  steelblue: "4682b4",
  tan: "d2b48c",
  teal: "008080",
  thistle: "d8bfd8",
  tomato: "ff6347",
  turquoise: "40e0d0",
  violet: "ee82ee",
  wheat: "f5deb3",
  white: "fff",
  whitesmoke: "f5f5f5",
  yellow: "ff0",
  yellowgreen: "9acd32"
};
var hexNames = tinycolor.hexNames = flip(names);
function flip(o) {
  var flipped = {};
  for (var i in o) {
    if (o.hasOwnProperty(i)) {
      flipped[o[i]] = i;
    }
  }
  return flipped;
}
function boundAlpha(a) {
  a = parseFloat(a);
  if (isNaN(a) || a < 0 || a > 1) {
    a = 1;
  }
  return a;
}
function bound01(n, max) {
  if (isOnePointZero(n)) n = "100%";
  var processPercent = isPercentage(n);
  n = Math.min(max, Math.max(0, parseFloat(n)));
  if (processPercent) {
    n = parseInt(n * max, 10) / 100;
  }
  if (Math.abs(n - max) < 1e-6) {
    return 1;
  }
  return n % max / parseFloat(max);
}
function clamp01(val) {
  return Math.min(1, Math.max(0, val));
}
function parseIntFromHex(val) {
  return parseInt(val, 16);
}
function isOnePointZero(n) {
  return typeof n == "string" && n.indexOf(".") != -1 && parseFloat(n) === 1;
}
function isPercentage(n) {
  return typeof n === "string" && n.indexOf("%") != -1;
}
function pad2(c) {
  return c.length == 1 ? "0" + c : "" + c;
}
function convertToPercentage(n) {
  if (n <= 1) {
    n = n * 100 + "%";
  }
  return n;
}
function convertDecimalToHex(d) {
  return Math.round(parseFloat(d) * 255).toString(16);
}
function convertHexToDecimal(h) {
  return parseIntFromHex(h) / 255;
}
var matchers = (function() {
  var CSS_INTEGER = "[-\\+]?\\d+%?";
  var CSS_NUMBER = "[-\\+]?\\d*\\.\\d+%?";
  var CSS_UNIT = "(?:" + CSS_NUMBER + ")|(?:" + CSS_INTEGER + ")";
  var PERMISSIVE_MATCH3 = "[\\s|\\(]+(" + CSS_UNIT + ")[,|\\s]+(" + CSS_UNIT + ")[,|\\s]+(" + CSS_UNIT + ")\\s*\\)?";
  var PERMISSIVE_MATCH4 = "[\\s|\\(]+(" + CSS_UNIT + ")[,|\\s]+(" + CSS_UNIT + ")[,|\\s]+(" + CSS_UNIT + ")[,|\\s]+(" + CSS_UNIT + ")\\s*\\)?";
  return {
    CSS_UNIT: new RegExp(CSS_UNIT),
    rgb: new RegExp("rgb" + PERMISSIVE_MATCH3),
    rgba: new RegExp("rgba" + PERMISSIVE_MATCH4),
    hsl: new RegExp("hsl" + PERMISSIVE_MATCH3),
    hsla: new RegExp("hsla" + PERMISSIVE_MATCH4),
    hsv: new RegExp("hsv" + PERMISSIVE_MATCH3),
    hsva: new RegExp("hsva" + PERMISSIVE_MATCH4),
    hex3: /^#?([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})$/,
    hex6: /^#?([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})$/,
    hex4: /^#?([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})$/,
    hex8: /^#?([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})$/
  };
})();
function isValidCSSUnit(color) {
  return !!matchers.CSS_UNIT.exec(color);
}
function stringInputToObject(color) {
  color = color.replace(trimLeft, "").replace(trimRight, "").toLowerCase();
  var named = false;
  if (names[color]) {
    color = names[color];
    named = true;
  } else if (color == "transparent") {
    return {
      r: 0,
      g: 0,
      b: 0,
      a: 0,
      format: "name"
    };
  }
  var match;
  if (match = matchers.rgb.exec(color)) {
    return {
      r: match[1],
      g: match[2],
      b: match[3]
    };
  }
  if (match = matchers.rgba.exec(color)) {
    return {
      r: match[1],
      g: match[2],
      b: match[3],
      a: match[4]
    };
  }
  if (match = matchers.hsl.exec(color)) {
    return {
      h: match[1],
      s: match[2],
      l: match[3]
    };
  }
  if (match = matchers.hsla.exec(color)) {
    return {
      h: match[1],
      s: match[2],
      l: match[3],
      a: match[4]
    };
  }
  if (match = matchers.hsv.exec(color)) {
    return {
      h: match[1],
      s: match[2],
      v: match[3]
    };
  }
  if (match = matchers.hsva.exec(color)) {
    return {
      h: match[1],
      s: match[2],
      v: match[3],
      a: match[4]
    };
  }
  if (match = matchers.hex8.exec(color)) {
    return {
      r: parseIntFromHex(match[1]),
      g: parseIntFromHex(match[2]),
      b: parseIntFromHex(match[3]),
      a: convertHexToDecimal(match[4]),
      format: named ? "name" : "hex8"
    };
  }
  if (match = matchers.hex6.exec(color)) {
    return {
      r: parseIntFromHex(match[1]),
      g: parseIntFromHex(match[2]),
      b: parseIntFromHex(match[3]),
      format: named ? "name" : "hex"
    };
  }
  if (match = matchers.hex4.exec(color)) {
    return {
      r: parseIntFromHex(match[1] + "" + match[1]),
      g: parseIntFromHex(match[2] + "" + match[2]),
      b: parseIntFromHex(match[3] + "" + match[3]),
      a: convertHexToDecimal(match[4] + "" + match[4]),
      format: named ? "name" : "hex8"
    };
  }
  if (match = matchers.hex3.exec(color)) {
    return {
      r: parseIntFromHex(match[1] + "" + match[1]),
      g: parseIntFromHex(match[2] + "" + match[2]),
      b: parseIntFromHex(match[3] + "" + match[3]),
      format: named ? "name" : "hex"
    };
  }
  return false;
}
function validateWCAG2Parms(parms) {
  var level, size;
  parms = parms || {
    level: "AA",
    size: "small"
  };
  level = (parms.level || "AA").toUpperCase();
  size = (parms.size || "small").toLowerCase();
  if (level !== "AA" && level !== "AAA") {
    level = "AA";
  }
  if (size !== "small" && size !== "large") {
    size = "small";
  }
  return {
    level,
    size
  };
}

// ../catppuccin-vsc/src/theme/utilities.ts
var mix = (color1, color2, amount) => tinycolor.mix(color1, color2, amount * 100).toHexString();
var opacity = (color, opacity2) => tinycolor(color).setAlpha(opacity2).toHex8String();
var shade = (color, magnitude) => magnitude > 0 ? tinycolor(color).lighten(magnitude * 100).toHexString() : tinycolor(color).darken(Math.abs(magnitude * 100)).toHexString();
var transparent = "#00000000";

// ../catppuccin-vsc/src/theme/extensions/error-lens.ts
function colors(context) {
  const { palette } = context;
  return {
    "errorLens.errorBackground": opacity(palette.red, 0.15),
    "errorLens.errorBackgroundLight": opacity(palette.red, 0.15),
    "errorLens.errorForeground": palette.red,
    "errorLens.errorForegroundLight": palette.red,
    "errorLens.errorMessageBackground": opacity(palette.red, 0.15),
    "errorLens.hintBackground": opacity(palette.green, 0.15),
    "errorLens.hintBackgroundLight": opacity(palette.green, 0.15),
    "errorLens.hintForeground": palette.green,
    "errorLens.hintForegroundLight": palette.green,
    "errorLens.hintMessageBackground": opacity(palette.green, 0.15),
    "errorLens.infoBackground": opacity(palette.blue, 0.15),
    "errorLens.infoBackgroundLight": opacity(palette.blue, 0.15),
    "errorLens.infoForeground": palette.blue,
    "errorLens.infoForegroundLight": palette.blue,
    "errorLens.infoMessageBackground": opacity(palette.blue, 0.15),
    "errorLens.statusBarErrorForeground": palette.red,
    "errorLens.statusBarHintForeground": palette.green,
    "errorLens.statusBarIconErrorForeground": palette.red,
    "errorLens.statusBarIconWarningForeground": palette.peach,
    "errorLens.statusBarInfoForeground": palette.blue,
    "errorLens.statusBarWarningForeground": palette.peach,
    "errorLens.warningBackground": opacity(palette.peach, 0.15),
    "errorLens.warningBackgroundLight": opacity(palette.peach, 0.15),
    "errorLens.warningForeground": palette.peach,
    "errorLens.warningForegroundLight": palette.peach,
    "errorLens.warningMessageBackground": opacity(palette.peach, 0.15)
  };
}

// ../catppuccin-vsc/src/theme/extensions/github-pull-request.ts
function colors2(context) {
  const { palette } = context;
  return {
    "issues.closed": palette.mauve,
    "issues.newIssueDecoration": palette.rosewater,
    "issues.open": palette.green,
    "pullRequests.closed": palette.red,
    "pullRequests.draft": palette.overlay2,
    "pullRequests.merged": palette.mauve,
    "pullRequests.notification": palette.text,
    "pullRequests.open": palette.green
  };
}

// ../catppuccin-vsc/src/theme/extensions/gitlens.ts
function colors3(context) {
  const { palette, options } = context;
  const graphLaneColors = [
    palette.mauve,
    palette.yellow,
    palette.blue,
    palette.flamingo,
    palette.green,
    palette.lavender,
    palette.rosewater,
    palette.red,
    palette.teal,
    palette.pink,
    palette.peach,
    palette.sapphire,
    palette.maroon,
    palette.sky
  ];
  const accentIndex = graphLaneColors.indexOf(palette[options.accent]);
  return {
    "gitlens.gutterBackgroundColor": opacity(palette.surface0, 0.3),
    "gitlens.gutterForegroundColor": palette.text,
    "gitlens.gutterUncommittedForegroundColor": palette[options.accent],
    "gitlens.trailingLineBackgroundColor": transparent,
    "gitlens.trailingLineForegroundColor": opacity(palette.text, 0.3),
    "gitlens.lineHighlightBackgroundColor": opacity(
      palette[options.accent],
      0.15
    ),
    "gitlens.lineHighlightOverviewRulerColor": opacity(
      palette[options.accent],
      0.8
    ),
    "gitlens.openAutolinkedIssueIconColor": palette.green,
    "gitlens.closedAutolinkedIssueIconColor": palette.mauve,
    "gitlens.closedPullRequestIconColor": palette.red,
    "gitlens.openPullRequestIconColor": palette.green,
    "gitlens.mergedPullRequestIconColor": palette.mauve,
    "gitlens.unpublishedChangesIconColor": palette.green,
    "gitlens.unpublishedCommitIconColor": palette.green,
    "gitlens.unpulledChangesIconColor": palette.peach,
    // "gitlens.decorations.addedForegroundColor": "", // (default: gitDecoration.addedResourceForeground)
    // "gitlens.decorations.copiedForegroundColor": "", // (default: gitDecoration.renamedResourceForeground)
    // "gitlens.decorations.deletedForegroundColor": "", // (default: gitDecoration.deletedResourceForeground)
    // "gitlens.decorations.ignoredForegroundColor": "", // (default: gitDecoration.ignoredResourceForeground)
    // "gitlens.decorations.modifiedForegroundColor": "", // (default: gitDecoration.modifiedResourceForeground)
    // "gitlens.decorations.untrackedForegroundColor": "", // (default: gitDecoration.untrackedResourceForeground)
    // "gitlens.decorations.renamedForegroundColor": "", // (default: gitDecoration.renamedResourceForeground)
    "gitlens.decorations.branchAheadForegroundColor": palette.green,
    "gitlens.decorations.branchBehindForegroundColor": palette.peach,
    "gitlens.decorations.branchDivergedForegroundColor": palette.yellow,
    // "gitlens.decorations.branchUpToDateForegroundColor": palette.text, // (default: sideBar.foreground)
    "gitlens.decorations.branchUnpublishedForegroundColor": palette.green,
    "gitlens.decorations.branchMissingUpstreamForegroundColor": palette.peach,
    "gitlens.decorations.statusMergingOrRebasingConflictForegroundColor": palette.maroon,
    "gitlens.decorations.statusMergingOrRebasingForegroundColor": palette.yellow,
    "gitlens.decorations.workspaceRepoMissingForegroundColor": palette.subtext0,
    "gitlens.decorations.workspaceCurrentForegroundColor": palette[options.accent],
    "gitlens.decorations.workspaceRepoOpenForegroundColor": palette[options.accent],
    "gitlens.decorations.worktreeHasUncommittedChangesForegroundColor": palette.peach,
    "gitlens.decorations.worktreeMissingForegroundColor": palette.maroon,
    "gitlens.graphLane1Color": graphLaneColors[accentIndex],
    "gitlens.graphLane2Color": graphLaneColors[(accentIndex + 1) % graphLaneColors.length],
    "gitlens.graphLane3Color": graphLaneColors[(accentIndex + 2) % graphLaneColors.length],
    "gitlens.graphLane4Color": graphLaneColors[(accentIndex + 3) % graphLaneColors.length],
    "gitlens.graphLane5Color": graphLaneColors[(accentIndex + 4) % graphLaneColors.length],
    "gitlens.graphLane6Color": graphLaneColors[(accentIndex + 5) % graphLaneColors.length],
    "gitlens.graphLane7Color": graphLaneColors[(accentIndex + 6) % graphLaneColors.length],
    "gitlens.graphLane8Color": graphLaneColors[(accentIndex + 7) % graphLaneColors.length],
    "gitlens.graphLane9Color": graphLaneColors[(accentIndex + 8) % graphLaneColors.length],
    "gitlens.graphLane10Color": graphLaneColors[(accentIndex + 9) % graphLaneColors.length],
    "gitlens.graphChangesColumnAddedColor": palette.green,
    "gitlens.graphChangesColumnDeletedColor": palette.red,
    "gitlens.graphMinimapMarkerHeadColor": palette.green,
    "gitlens.graphScrollMarkerHeadColor": palette.green,
    "gitlens.graphMinimapMarkerUpstreamColor": shade(palette.green, -0.05),
    "gitlens.graphScrollMarkerUpstreamColor": shade(palette.green, -0.05),
    "gitlens.graphMinimapMarkerHighlightsColor": palette.yellow,
    "gitlens.graphScrollMarkerHighlightsColor": palette.yellow,
    "gitlens.graphMinimapMarkerLocalBranchesColor": palette.blue,
    "gitlens.graphScrollMarkerLocalBranchesColor": palette.blue,
    "gitlens.graphMinimapMarkerRemoteBranchesColor": shade(palette.blue, -0.05),
    "gitlens.graphScrollMarkerRemoteBranchesColor": shade(palette.blue, -0.05),
    "gitlens.graphMinimapMarkerStashesColor": palette.mauve,
    "gitlens.graphScrollMarkerStashesColor": palette.mauve,
    "gitlens.graphMinimapMarkerTagsColor": palette.flamingo,
    "gitlens.graphScrollMarkerTagsColor": palette.flamingo
  };
}

// ../catppuccin-vsc/src/theme/extensions/index.ts
function extensions(context) {
  return {
    ...colors(context),
    ...colors2(context),
    ...colors3(context)
  };
}

// ../catppuccin-vsc/src/theme/ui/azureDataStudio.ts
var azureDataStudio = (context) => {
  const { options, palette } = context;
  const accent = palette[options.accent];
  return {
    "button.secondaryBorder": accent,
    "table.headerBackground": palette.surface0,
    "table.headerForeground": palette.text,
    "list.focusAndSelectionBackground": palette.surface1
  };
};
var azureDataStudio_default = azureDataStudio;

// ../catppuccin-vsc/src/theme/ui/brackets.ts
var brackets = (context) => {
  const { isLatte, options, palette } = context;
  const dimAmount = -0.07 * (isLatte ? -1 : 1);
  const styles = {
    rainbow: {
      "editorBracketHighlight.foreground1": palette.red,
      "editorBracketHighlight.foreground2": palette.peach,
      "editorBracketHighlight.foreground3": palette.yellow,
      "editorBracketHighlight.foreground4": palette.green,
      "editorBracketHighlight.foreground5": palette.sapphire,
      "editorBracketHighlight.foreground6": palette.mauve,
      "editorBracketHighlight.unexpectedBracket.foreground": palette.maroon
    },
    dimmed: {
      "editorBracketHighlight.foreground1": shade(palette.red, dimAmount),
      "editorBracketHighlight.foreground2": shade(palette.peach, dimAmount),
      "editorBracketHighlight.foreground3": shade(palette.yellow, dimAmount),
      "editorBracketHighlight.foreground4": shade(palette.green, dimAmount),
      "editorBracketHighlight.foreground5": shade(palette.sapphire, dimAmount),
      "editorBracketHighlight.foreground6": shade(palette.mauve, dimAmount),
      "editorBracketHighlight.unexpectedBracket.foreground": shade(
        palette.maroon,
        dimAmount
      )
    },
    monochromatic: {
      "editorBracketHighlight.foreground1": palette.subtext1,
      "editorBracketHighlight.foreground2": palette.subtext0,
      "editorBracketHighlight.foreground3": palette.overlay2,
      "editorBracketHighlight.foreground4": palette.overlay1,
      "editorBracketHighlight.foreground5": palette.overlay0,
      "editorBracketHighlight.foreground6": palette.surface2,
      "editorBracketHighlight.unexpectedBracket.foreground": palette.maroon
    },
    neovim: {
      "editorBracketHighlight.foreground1": palette.red,
      "editorBracketHighlight.foreground2": palette.teal,
      "editorBracketHighlight.foreground3": palette.yellow,
      "editorBracketHighlight.foreground4": palette.blue,
      "editorBracketHighlight.foreground5": palette.pink,
      "editorBracketHighlight.foreground6": palette.flamingo,
      "editorBracketHighlight.unexpectedBracket.foreground": palette.maroon
    }
  };
  return styles[options.bracketMode];
};
var brackets_default = brackets;

// ../catppuccin-vsc/src/theme/ui/workbench.ts
var workbench = (context) => {
  const { options, palette } = context;
  const styles = {
    default: {},
    flat: {
      "activityBar.background": palette.mantle,
      "breadcrumb.background": palette.base,
      "commandCenter.background": palette.mantle,
      "debugToolBar.background": palette.mantle,
      "editorGroupHeader.tabsBackground": palette.mantle,
      "minimap.background": opacity(palette.base, 0.5),
      "statusBar.background": palette.mantle,
      "statusBar.noFolderBackground": palette.mantle,
      "tab.border": palette.base,
      "titleBar.activeBackground": palette.mantle,
      "titleBar.inactiveBackground": palette.mantle,
      "scrollbar.shadow": palette.mantle
    },
    minimal: {
      "activityBar.background": palette.base,
      "breadcrumb.background": palette.base,
      "commandCenter.background": palette.base,
      "debugToolBar.background": palette.base,
      "editor.background": palette.base,
      "editorWidget.background": palette.base,
      "editorGroupHeader.tabsBackground": palette.base,
      "minimap.background": opacity(palette.base, 0.5),
      "statusBar.background": palette.base,
      "statusBar.noFolderBackground": palette.base,
      "sideBar.background": palette.base,
      "sideBarSectionHeader.background": palette.base,
      "tab.border": palette.base,
      "tab.inactiveBackground": palette.base,
      "titleBar.activeBackground": palette.base,
      "titleBar.inactiveBackground": palette.base
    }
  };
  return styles[options.workbenchMode];
};
var workbench_default = workbench;

// ../catppuccin-vsc/src/theme/ui/customNames.ts
var import_palette = require("@catppuccin/palette");
var ctpColors = new Set(Object.keys(import_palette.flavors.mocha.colors));
var CustomUIColorError = class extends Error {
  constructor(key, value, message = "") {
    super(
      `Invalid value: "${value}" for key "${key}" in "catppuccin.customUIColors".
${message}`
    );
    this.name = "CustomUIColorError";
  }
};
var parseCustomUiColor = (k, v) => {
  const entry = v.split(" ");
  if (entry.length > 2) {
    throw new CustomUIColorError(
      k,
      v,
      'Too many arguments, expected "colorName" or "colorName opacityValue".'
    );
  }
  let opacityValue = 1;
  if (entry.length == 2) {
    opacityValue = Number(entry[1]);
    if (Number.isNaN(opacityValue)) {
      throw new CustomUIColorError(k, v, "Opacity value is not a number.");
    }
    if (opacityValue < 0 || opacityValue > 1) {
      throw new CustomUIColorError(
        k,
        v,
        "Opacity value is not between 0 and 1."
      );
    }
  }
  return [entry[0], opacityValue];
};
var hexColorRegex = /^#([\dA-Fa-f]{3,4}){1,2}$/;
var customNamedColors = (context) => {
  const { flavor, palette, options } = context;
  const accent = palette[options.accent];
  const customUIColors = {
    ...options.customUIColors.all,
    ...options.customUIColors[flavor]
  };
  for (const [k, v] of Object.entries(customUIColors)) {
    if (hexColorRegex.test(v)) continue;
    const [parsedColor, opacityValue] = parseCustomUiColor(k, v);
    let color;
    if (parsedColor === "accent") {
      color = accent;
    } else if (ctpColors.has(parsedColor)) {
      color = palette[parsedColor];
    } else {
      throw new CustomUIColorError(k, v, "Invalid color name.");
    }
    customUIColors[k] = opacity(color, opacityValue);
  }
  return customUIColors;
};
var customNames_default = customNamedColors;

// ../catppuccin-vsc/src/theme/ui/index.ts
var uiCustomizations = (context) => {
  return {
    ...brackets_default(context),
    ...workbench_default(context),
    ...customNames_default(context),
    ...azureDataStudio_default(context)
  };
};
var ui_default = uiCustomizations;

// ../catppuccin-vsc/src/theme/uiColors.ts
var getUiColors = (context) => {
  const { palette, paletteAnsi, options, isLatte } = context;
  const accent = palette[options.accent];
  const dropBackground = opacity(accent, 0.2);
  const border = options.extraBordersEnabled ? opacity(palette.overlay1, 0.15) : transparent;
  return {
    // Base colors
    focusBorder: accent,
    foreground: palette.text,
    disabledForeground: palette.subtext0,
    "widget.shadow": opacity(palette.mantle, 0.5),
    "selection.background": opacity(accent, 0.4),
    descriptionForeground: palette.text,
    errorForeground: palette.red,
    "icon.foreground": accent,
    "sash.hoverBorder": accent,
    // Window border
    "window.activeBorder": transparent,
    "window.inactiveBorder": transparent,
    // Text colors
    "textBlockQuote.background": palette.mantle,
    "textBlockQuote.border": palette.crust,
    "textCodeBlock.background": palette.mantle,
    "textLink.activeForeground": palette.sky,
    "textLink.foreground": palette.blue,
    "textPreformat.foreground": palette.text,
    "textSeparator.foreground": accent,
    // Activity Bar
    "activityBar.background": palette.crust,
    "activityBar.foreground": accent,
    "activityBar.dropBorder": dropBackground,
    "activityBar.inactiveForeground": palette.overlay0,
    "activityBar.border": border,
    "activityBarBadge.background": accent,
    "activityBarBadge.foreground": palette.crust,
    "activityBar.activeBorder": transparent,
    "activityBar.activeBackground": transparent,
    "activityBar.activeFocusBorder": transparent,
    "activityBarTop.foreground": accent,
    "activityBarTop.activeBorder": transparent,
    "activityBarTop.inactiveForeground": palette.overlay0,
    "activityBarTop.dropBorder": dropBackground,
    "badge.background": palette.surface1,
    "badge.foreground": palette.text,
    "breadcrumb.activeSelectionForeground": accent,
    "breadcrumb.background": palette.base,
    "breadcrumb.focusForeground": accent,
    "breadcrumb.foreground": opacity(palette.text, 0.8),
    "breadcrumbPicker.background": palette.mantle,
    // buttons & checkboxes
    "button.background": accent,
    "button.foreground": palette.crust,
    "button.border": transparent,
    "button.separator": transparent,
    "button.hoverBackground": shade(accent, 0.07),
    "button.secondaryForeground": palette.text,
    "button.secondaryBackground": palette.surface2,
    "button.secondaryHoverBackground": shade(palette.surface2, 0.07),
    "checkbox.background": palette.surface1,
    "checkbox.border": border,
    "checkbox.foreground": accent,
    // dropdown controls
    "dropdown.background": palette.mantle,
    "dropdown.listBackground": palette.surface2,
    "dropdown.border": accent,
    "dropdown.foreground": palette.text,
    // debug
    "debugToolBar.background": palette.crust,
    "debugToolBar.border": transparent,
    "debugExceptionWidget.background": palette.crust,
    "debugExceptionWidget.border": accent,
    "debugTokenExpression.number": palette.peach,
    "debugTokenExpression.boolean": palette.mauve,
    "debugTokenExpression.string": palette.green,
    "debugTokenExpression.error": palette.red,
    // debug icons
    "debugIcon.breakpointForeground": palette.red,
    "debugIcon.breakpointDisabledForeground": opacity(palette.red, 0.6),
    "debugIcon.breakpointUnverifiedForeground": mix(
      palette.red,
      palette.surface2,
      0.5
    ),
    "debugIcon.breakpointCurrentStackframeForeground": palette.surface2,
    "debugIcon.breakpointStackframeForeground": palette.surface2,
    "debugIcon.startForeground": palette.green,
    "debugIcon.pauseForeground": palette.blue,
    "debugIcon.stopForeground": palette.red,
    "debugIcon.disconnectForeground": palette.surface2,
    "debugIcon.restartForeground": palette.teal,
    "debugIcon.stepOverForeground": palette.mauve,
    "debugIcon.stepIntoForeground": palette.text,
    "debugIcon.stepOutForeground": palette.text,
    "debugIcon.continueForeground": palette.green,
    "debugIcon.stepBackForeground": palette.surface2,
    "debugConsole.infoForeground": palette.blue,
    "debugConsole.warningForeground": palette.peach,
    "debugConsole.errorForeground": palette.red,
    "debugConsole.sourceForeground": palette.rosewater,
    "debugConsoleInputIcon.foreground": palette.text,
    // testing
    "testing.runAction": accent,
    "testing.iconErrored": palette.red,
    "testing.iconFailed": palette.red,
    "testing.iconPassed": palette.green,
    "testing.iconQueued": palette.blue,
    "testing.iconUnset": palette.text,
    "testing.iconSkipped": palette.subtext0,
    "testing.iconErrored.retired": palette.red,
    "testing.iconFailed.retired": palette.red,
    "testing.iconPassed.retired": palette.green,
    "testing.iconQueued.retired": palette.blue,
    "testing.iconUnset.retired": palette.text,
    "testing.iconSkipped.retired": palette.subtext0,
    "testing.peekBorder": accent,
    "testing.peekHeaderBackground": palette.surface2,
    "testing.message.error.lineBackground": opacity(palette.red, 0.15),
    "testing.message.info.decorationForeground": opacity(palette.green, 0.8),
    "testing.message.info.lineBackground": opacity(palette.green, 0.15),
    "testing.messagePeekBorder": accent,
    "testing.messagePeekHeaderBackground": palette.surface2,
    "testing.coveredBackground": opacity(palette.green, 0.3),
    "testing.coveredBorder": transparent,
    "testing.coveredGutterBackground": opacity(palette.green, 0.3),
    "testing.uncoveredBranchBackground": opacity(palette.red, 0.2),
    "testing.uncoveredBackground": opacity(palette.red, 0.2),
    "testing.uncoveredBorder": transparent,
    "testing.uncoveredGutterBackground": opacity(palette.red, 0.25),
    "testing.coverCountBadgeBackground": transparent,
    "testing.coverCountBadgeForeground": accent,
    "diffEditor.border": palette.surface2,
    "diffEditor.insertedTextBackground": opacity(palette.green, 0.2),
    "diffEditor.removedTextBackground": opacity(palette.red, 0.2),
    "diffEditor.insertedLineBackground": opacity(palette.green, 0.15),
    "diffEditor.removedLineBackground": opacity(palette.red, 0.15),
    "diffEditor.diagonalFill": opacity(palette.surface2, 0.6),
    "diffEditorOverview.insertedForeground": opacity(palette.green, 0.8),
    "diffEditorOverview.removedForeground": opacity(palette.red, 0.8),
    "editor.background": palette.base,
    "editor.findMatchBackground": mix(palette.base, palette.red, 0.3),
    "editor.findMatchBorder": opacity(palette.red, 0.2),
    "editor.findMatchHighlightBackground": mix(palette.base, palette.sky, 0.3),
    "editor.findMatchHighlightBorder": opacity(palette.sky, 0.2),
    "editor.findRangeHighlightBackground": mix(palette.base, palette.sky, 0.3),
    "editor.findRangeHighlightBorder": opacity(palette.sky, 0.2),
    "editor.foldBackground": opacity(palette.sky, 0.25),
    "editor.foreground": palette.text,
    "editor.hoverHighlightBackground": opacity(palette.sky, 0.25),
    "editor.lineHighlightBackground": opacity(palette.text, 0.07),
    "editor.lineHighlightBorder": transparent,
    "editor.rangeHighlightBackground": opacity(palette.sky, 0.25),
    "editor.rangeHighlightBorder": transparent,
    "editor.selectionBackground": opacity(
      palette.overlay2,
      isLatte ? 0.3 : 0.25
    ),
    "editor.selectionHighlightBackground": opacity(palette.overlay2, 0.2),
    "editor.selectionHighlightBorder": opacity(palette.overlay2, 0.2),
    "editor.wordHighlightBackground": opacity(palette.overlay2, 0.2),
    "editor.wordHighlightStrongBackground": opacity(
      palette.blue,
      isLatte ? 0.15 : 0.2
    ),
    "editorBracketMatch.background": opacity(palette.overlay2, 0.1),
    "editorBracketMatch.border": palette.overlay2,
    "editorCodeLens.foreground": palette.overlay1,
    "editorCursor.background": palette.base,
    "editorCursor.foreground": palette.rosewater,
    "editorGroup.border": palette.surface2,
    "editorGroup.dropBackground": dropBackground,
    "editorGroup.emptyBackground": palette.base,
    "editorGroupHeader.tabsBackground": palette.crust,
    "editorGutter.addedBackground": palette.green,
    "editorGutter.background": palette.base,
    "editorGutter.commentRangeForeground": palette.surface0,
    "editorGutter.commentGlyphForeground": accent,
    "editorGutter.deletedBackground": palette.red,
    "editorGutter.foldingControlForeground": palette.overlay2,
    "editorGutter.modifiedBackground": palette.yellow,
    "editorHoverWidget.background": palette.mantle,
    "editorHoverWidget.border": palette.surface2,
    "editorHoverWidget.foreground": palette.text,
    "editorIndentGuide.activeBackground": palette.surface2,
    "editorIndentGuide.background": palette.surface1,
    "editorInlayHint.foreground": palette.surface2,
    "editorInlayHint.background": opacity(palette.mantle, 0.75),
    "editorInlayHint.typeForeground": palette.subtext1,
    "editorInlayHint.typeBackground": opacity(palette.mantle, 0.75),
    "editorInlayHint.parameterForeground": palette.subtext0,
    "editorInlayHint.parameterBackground": opacity(palette.mantle, 0.75),
    "editorLineNumber.activeForeground": accent,
    "editorLineNumber.foreground": palette.overlay1,
    "editorLink.activeForeground": accent,
    "editorMarkerNavigation.background": palette.mantle,
    "editorMarkerNavigationError.background": palette.red,
    "editorMarkerNavigationInfo.background": palette.blue,
    "editorMarkerNavigationWarning.background": palette.peach,
    "editorOverviewRuler.background": palette.mantle,
    "editorOverviewRuler.border": opacity(palette.text, 0.07),
    "editorOverviewRuler.modifiedForeground": palette.yellow,
    "editorRuler.foreground": palette.surface2,
    // breakpoints
    "editor.stackFrameHighlightBackground": opacity(palette.yellow, 0.15),
    "editor.focusedStackFrameHighlightBackground": opacity(palette.green, 0.15),
    "editorStickyScrollHover.background": palette.surface0,
    "editorSuggestWidget.background": palette.mantle,
    "editorSuggestWidget.border": palette.surface2,
    "editorSuggestWidget.foreground": palette.text,
    "editorSuggestWidget.highlightForeground": accent,
    "editorSuggestWidget.selectedBackground": palette.surface0,
    "editorWhitespace.foreground": opacity(palette.overlay2, 0.4),
    "editorWidget.background": palette.mantle,
    "editorWidget.foreground": palette.text,
    "editorWidget.resizeBorder": palette.surface2,
    "editorLightBulb.foreground": palette.yellow,
    // errors & warnings
    "editorError.foreground": palette.red,
    "editorError.border": transparent,
    "editorError.background": transparent,
    "editorWarning.foreground": palette.peach,
    "editorWarning.border": transparent,
    "editorWarning.background": transparent,
    "editorInfo.foreground": palette.blue,
    "editorInfo.border": transparent,
    "editorInfo.background": transparent,
    "problemsErrorIcon.foreground": palette.red,
    "problemsInfoIcon.foreground": palette.blue,
    "problemsWarningIcon.foreground": palette.peach,
    // extensions marketplace
    "extensionButton.prominentForeground": palette.crust,
    "extensionButton.prominentBackground": accent,
    "extensionButton.separator": palette.base,
    "extensionButton.prominentHoverBackground": shade(accent, 0.07),
    "extensionBadge.remoteBackground": palette.blue,
    "extensionBadge.remoteForeground": palette.crust,
    "extensionIcon.starForeground": palette.yellow,
    "extensionIcon.verifiedForeground": palette.green,
    "extensionIcon.preReleaseForeground": palette.surface2,
    "extensionIcon.sponsorForeground": palette.pink,
    // git colors
    "gitDecoration.addedResourceForeground": palette.green,
    "gitDecoration.conflictingResourceForeground": palette.mauve,
    "gitDecoration.deletedResourceForeground": palette.red,
    "gitDecoration.ignoredResourceForeground": palette.overlay0,
    "gitDecoration.modifiedResourceForeground": palette.yellow,
    "gitDecoration.stageDeletedResourceForeground": palette.red,
    "gitDecoration.stageModifiedResourceForeground": palette.yellow,
    "gitDecoration.submoduleResourceForeground": palette.blue,
    "gitDecoration.untrackedResourceForeground": palette.green,
    // git colors -> built-in git graph
    "scmGraph.historyItemRefColor": palette.blue,
    "scmGraph.historyItemBaseRefColor": palette.peach,
    "scmGraph.historyItemRemoteRefColor": palette.mauve,
    "scmGraph.foreground1": palette.yellow,
    "scmGraph.foreground2": palette.red,
    "scmGraph.foreground3": palette.green,
    "scmGraph.foreground4": palette.mauve,
    "scmGraph.foreground5": palette.teal,
    "input.background": palette.surface0,
    "input.border": transparent,
    "input.foreground": palette.text,
    "input.placeholderForeground": opacity(palette.text, 0.45),
    "inputOption.activeBackground": palette.surface2,
    "inputOption.activeBorder": accent,
    "inputOption.activeForeground": palette.text,
    "inputValidation.errorBackground": palette.red,
    "inputValidation.errorBorder": opacity(palette.crust, 0.2),
    "inputValidation.errorForeground": palette.crust,
    "inputValidation.infoBackground": palette.blue,
    "inputValidation.infoBorder": opacity(palette.crust, 0.2),
    "inputValidation.infoForeground": palette.crust,
    "inputValidation.warningBackground": palette.peach,
    "inputValidation.warningBorder": opacity(palette.crust, 0.2),
    "inputValidation.warningForeground": palette.crust,
    // Lists and trees
    "list.activeSelectionBackground": palette.surface0,
    // currently selected in file tree
    "list.activeSelectionForeground": palette.text,
    "list.dropBackground": dropBackground,
    "list.focusBackground": palette.surface0,
    // when using keyboard to move around files
    "list.focusForeground": palette.text,
    "list.focusOutline": transparent,
    "list.highlightForeground": accent,
    "list.hoverBackground": opacity(palette.surface0, 0.5),
    // when hovering over the file tree
    "list.hoverForeground": palette.text,
    "list.inactiveSelectionBackground": palette.surface0,
    // currently selected focused in editor
    "list.inactiveSelectionForeground": palette.text,
    "list.warningForeground": palette.peach,
    "listFilterWidget.background": palette.surface1,
    "listFilterWidget.noMatchesOutline": palette.red,
    "listFilterWidget.outline": transparent,
    "tree.indentGuidesStroke": palette.overlay2,
    "tree.inactiveIndentGuidesStroke": palette.surface1,
    "menu.background": palette.base,
    "menu.border": options.extraBordersEnabled ? palette.surface2 : opacity(palette.base, 0.5),
    "menu.foreground": palette.text,
    "menu.selectionBackground": palette.surface2,
    "menu.selectionBorder": transparent,
    "menu.selectionForeground": palette.text,
    "menu.separatorBackground": palette.surface2,
    "menubar.selectionBackground": palette.surface1,
    "menubar.selectionForeground": palette.text,
    "merge.commonContentBackground": palette.surface1,
    "merge.commonHeaderBackground": palette.surface2,
    "merge.currentContentBackground": opacity(palette.green, 0.2),
    "merge.currentHeaderBackground": opacity(palette.green, 0.4),
    "merge.incomingContentBackground": opacity(palette.blue, 0.2),
    "merge.incomingHeaderBackground": opacity(palette.blue, 0.4),
    "minimap.background": opacity(palette.mantle, 0.5),
    "minimap.findMatchHighlight": opacity(palette.sky, 0.3),
    "minimap.selectionHighlight": opacity(palette.surface2, 0.75),
    "minimap.selectionOccurrenceHighlight": opacity(palette.surface2, 0.75),
    "minimap.warningHighlight": opacity(palette.peach, 0.75),
    "minimap.errorHighlight": opacity(palette.red, 0.75),
    "minimapSlider.background": opacity(accent, 0.2),
    "minimapSlider.hoverBackground": opacity(accent, 0.4),
    "minimapSlider.activeBackground": opacity(accent, 0.6),
    "minimapGutter.addedBackground": opacity(palette.green, 0.75),
    "minimapGutter.deletedBackground": opacity(palette.red, 0.75),
    "minimapGutter.modifiedBackground": opacity(palette.yellow, 0.75),
    "notificationCenter.border": accent,
    "notificationCenterHeader.foreground": palette.text,
    "notificationCenterHeader.background": palette.mantle,
    "notificationToast.border": accent,
    "notifications.foreground": palette.text,
    "notifications.background": palette.mantle,
    "notifications.border": accent,
    "notificationLink.foreground": palette.blue,
    "notificationsErrorIcon.foreground": palette.red,
    "notificationsWarningIcon.foreground": palette.peach,
    "notificationsInfoIcon.foreground": palette.blue,
    "panel.background": palette.base,
    "panel.border": palette.surface2,
    "panelSection.border": palette.surface2,
    "panelSection.dropBackground": dropBackground,
    "panelTitle.activeBorder": accent,
    "panelTitle.activeForeground": palette.text,
    "panelTitle.inactiveForeground": palette.subtext0,
    // peek view colors
    "peekView.border": accent,
    "peekViewEditor.background": palette.mantle,
    "peekViewEditorGutter.background": palette.mantle,
    "peekViewEditor.matchHighlightBackground": opacity(palette.sky, 0.3),
    "peekViewEditor.matchHighlightBorder": transparent,
    "peekViewResult.background": palette.mantle,
    "peekViewResult.fileForeground": palette.text,
    "peekViewResult.lineForeground": palette.text,
    "peekViewResult.matchHighlightBackground": opacity(palette.sky, 0.3),
    "peekViewResult.selectionBackground": palette.surface0,
    "peekViewResult.selectionForeground": palette.text,
    "peekViewTitle.background": palette.base,
    "peekViewTitleDescription.foreground": opacity(palette.subtext1, 0.7),
    "peekViewTitleLabel.foreground": palette.text,
    "pickerGroup.border": accent,
    "pickerGroup.foreground": accent,
    "progressBar.background": accent,
    "scrollbar.shadow": palette.crust,
    "scrollbarSlider.activeBackground": opacity(palette.surface0, 0.4),
    "scrollbarSlider.background": opacity(palette.surface2, 0.5),
    "scrollbarSlider.hoverBackground": palette.overlay0,
    "settings.focusedRowBackground": opacity(palette.surface2, 0.2),
    "settings.headerForeground": palette.text,
    "settings.modifiedItemIndicator": accent,
    "settings.dropdownBackground": palette.surface1,
    "settings.dropdownListBorder": transparent,
    "settings.textInputBackground": palette.surface1,
    "settings.textInputBorder": transparent,
    "settings.numberInputBackground": palette.surface1,
    "settings.numberInputBorder": transparent,
    "sideBar.background": palette.mantle,
    "sideBar.dropBackground": dropBackground,
    "sideBar.foreground": palette.text,
    "sideBar.border": border,
    "sideBarSectionHeader.background": palette.mantle,
    "sideBarSectionHeader.foreground": palette.text,
    "sideBarTitle.foreground": accent,
    // banners, such as Restricted Mode
    "banner.background": palette.surface1,
    "banner.foreground": palette.text,
    "banner.iconForeground": palette.text,
    // Status Bar
    "statusBar.background": palette.crust,
    "statusBar.foreground": palette.text,
    "statusBar.border": border,
    // having no folder open shouldn't change the bar
    "statusBar.noFolderBackground": palette.crust,
    "statusBar.noFolderForeground": palette.text,
    "statusBar.noFolderBorder": border,
    // debugging is peach
    "statusBar.debuggingBackground": palette.peach,
    "statusBar.debuggingForeground": palette.crust,
    "statusBar.debuggingBorder": border,
    // remote is blue
    "statusBarItem.remoteBackground": palette.blue,
    "statusBarItem.remoteForeground": palette.crust,
    // different states
    "statusBarItem.activeBackground": opacity(palette.surface2, 0.4),
    "statusBarItem.hoverBackground": opacity(palette.surface2, 0.2),
    "statusBarItem.prominentForeground": accent,
    "statusBarItem.prominentBackground": transparent,
    "statusBarItem.prominentHoverBackground": opacity(palette.surface2, 0.2),
    "statusBarItem.errorForeground": palette.red,
    "statusBarItem.errorBackground": transparent,
    "statusBarItem.warningForeground": palette.peach,
    "statusBarItem.warningBackground": transparent,
    // command center
    "commandCenter.foreground": palette.subtext1,
    "commandCenter.inactiveForeground": palette.subtext1,
    "commandCenter.activeForeground": accent,
    "commandCenter.background": palette.mantle,
    "commandCenter.activeBackground": opacity(palette.surface2, 0.2),
    "commandCenter.border": border,
    "commandCenter.inactiveBorder": border,
    "commandCenter.activeBorder": accent,
    // Tab Bar
    "tab.activeBackground": palette.base,
    "tab.activeBorder": transparent,
    "tab.activeBorderTop": accent,
    "tab.activeForeground": accent,
    "tab.activeModifiedBorder": palette.yellow,
    "tab.border": palette.mantle,
    "tab.hoverBackground": shade(palette.base, 0.05),
    "tab.hoverBorder": transparent,
    "tab.hoverForeground": accent,
    "tab.inactiveBackground": palette.mantle,
    "tab.inactiveForeground": palette.overlay0,
    "tab.inactiveModifiedBorder": opacity(palette.yellow, 0.3),
    "tab.lastPinnedBorder": accent,
    "tab.unfocusedActiveBackground": palette.mantle,
    "tab.unfocusedActiveBorder": transparent,
    "tab.unfocusedActiveBorderTop": opacity(accent, 0.3),
    "tab.unfocusedInactiveBackground": shade(palette.mantle, -0.05),
    // Terminal
    "terminal.foreground": palette.text,
    "terminal.ansiBlack": paletteAnsi.normal.black,
    // color0
    "terminal.ansiRed": paletteAnsi.normal.red,
    // color1
    "terminal.ansiGreen": paletteAnsi.normal.green,
    // color2
    "terminal.ansiYellow": paletteAnsi.normal.yellow,
    // color3
    "terminal.ansiBlue": paletteAnsi.normal.blue,
    // color4
    "terminal.ansiMagenta": paletteAnsi.normal.magenta,
    // color5
    "terminal.ansiCyan": paletteAnsi.normal.cyan,
    // color6
    "terminal.ansiWhite": paletteAnsi.normal.white,
    // color7
    "terminal.ansiBrightBlack": paletteAnsi.bright.black,
    // color8
    "terminal.ansiBrightRed": paletteAnsi.bright.red,
    // color9
    "terminal.ansiBrightGreen": paletteAnsi.bright.green,
    // color10
    "terminal.ansiBrightYellow": paletteAnsi.bright.yellow,
    // color11
    "terminal.ansiBrightBlue": paletteAnsi.bright.blue,
    // color12
    "terminal.ansiBrightMagenta": paletteAnsi.bright.magenta,
    // color13
    "terminal.ansiBrightCyan": paletteAnsi.bright.cyan,
    // color14
    "terminal.ansiBrightWhite": paletteAnsi.bright.white,
    // color15
    "terminal.selectionBackground": palette.surface2,
    "terminal.inactiveSelectionBackground": opacity(palette.surface2, 0.5),
    "terminalCursor.background": palette.base,
    "terminalCursor.foreground": palette.rosewater,
    "terminal.border": palette.surface2,
    "terminal.dropBackground": dropBackground,
    "terminal.tab.activeBorder": accent,
    "terminalCommandDecoration.defaultBackground": palette.surface2,
    "terminalCommandDecoration.successBackground": palette.green,
    "terminalCommandDecoration.errorBackground": palette.red,
    // title bar
    "titleBar.activeBackground": palette.crust,
    "titleBar.activeForeground": palette.text,
    "titleBar.inactiveBackground": palette.crust,
    "titleBar.inactiveForeground": opacity(palette.text, 0.5),
    "titleBar.border": border,
    // welcome page
    "welcomePage.tileBackground": palette.mantle,
    "welcomePage.progress.background": palette.crust,
    "welcomePage.progress.foreground": accent,
    "walkThrough.embeddedEditorBackground": opacity(palette.base, 0.3),
    // symbols in outline, autocomplete, etc.
    "symbolIcon.textForeground": palette.text,
    "symbolIcon.arrayForeground": palette.peach,
    "symbolIcon.booleanForeground": palette.mauve,
    "symbolIcon.classForeground": palette.yellow,
    "symbolIcon.colorForeground": palette.pink,
    "symbolIcon.constantForeground": palette.peach,
    "symbolIcon.constructorForeground": palette.lavender,
    "symbolIcon.enumeratorForeground": palette.yellow,
    "symbolIcon.enumeratorMemberForeground": palette.yellow,
    "symbolIcon.eventForeground": palette.pink,
    "symbolIcon.fieldForeground": palette.text,
    "symbolIcon.fileForeground": accent,
    "symbolIcon.folderForeground": accent,
    "symbolIcon.functionForeground": palette.blue,
    "symbolIcon.interfaceForeground": palette.yellow,
    "symbolIcon.keyForeground": palette.teal,
    "symbolIcon.keywordForeground": palette.mauve,
    "symbolIcon.methodForeground": palette.blue,
    "symbolIcon.moduleForeground": palette.text,
    "symbolIcon.namespaceForeground": palette.yellow,
    "symbolIcon.nullForeground": palette.maroon,
    "symbolIcon.numberForeground": palette.peach,
    "symbolIcon.objectForeground": palette.yellow,
    "symbolIcon.operatorForeground": palette.teal,
    "symbolIcon.packageForeground": palette.flamingo,
    "symbolIcon.propertyForeground": palette.maroon,
    "symbolIcon.referenceForeground": palette.yellow,
    "symbolIcon.snippetForeground": palette.flamingo,
    "symbolIcon.stringForeground": palette.green,
    "symbolIcon.structForeground": palette.teal,
    "symbolIcon.typeParameterForeground": palette.maroon,
    "symbolIcon.unitForeground": palette.text,
    "symbolIcon.variableForeground": palette.text,
    // chart colors
    "charts.foreground": palette.text,
    "charts.lines": palette.subtext1,
    "charts.red": palette.red,
    "charts.blue": palette.blue,
    "charts.yellow": palette.yellow,
    "charts.orange": palette.peach,
    "charts.green": palette.green,
    "charts.purple": palette.mauve,
    ...extensions(context),
    ...ui_default(context)
  };
};

// ../catppuccin-vsc/src/theme/index.ts
var defaultOptions = {
  accent: "mauve",
  boldKeywords: true,
  italicComments: true,
  italicKeywords: true,
  colorOverrides: {},
  workbenchMode: "default",
  bracketMode: "rainbow",
  extraBordersEnabled: false,
  customUIColors: {},
  syncWithIconPack: true
};
var compileTheme = (flavor = "mocha", options = defaultOptions) => {
  var _a, _b;
  const flavorData = import_palette2.flavors[flavor];
  const ctpPalette = {};
  const paletteAnsi = {
    normal: {},
    bright: {}
  };
  for (const [k, v] of flavorData.colorEntries) {
    ctpPalette[k] = v.hex;
  }
  for (const [k, v] of flavorData.ansiColorEntries) {
    paletteAnsi.normal[k] = v.normal.hex;
    paletteAnsi.bright[k] = v.bright.hex;
  }
  const palette = {
    ...ctpPalette,
    ...(_a = options.colorOverrides) == null ? void 0 : _a.all,
    ...(_b = options.colorOverrides) == null ? void 0 : _b[flavor]
  };
  const context = {
    flavor,
    palette,
    paletteAnsi,
    options,
    isLatte: flavor === "latte"
  };
  return {
    name: `Catppuccin ${flavorData.name}`,
    type: context.isLatte ? "light" : "dark",
    colors: getUiColors(context),
    semanticHighlighting: true,
    semanticTokenColors: getSemanticTokens(context),
    tokenColors: getTokenColors(context)
  };
};

// src/compile.ts
var compile = (flavor, overrides = {}) => {
  const options = {
    ...defaultOptions,
    ...overrides
  };
  const compiled = compileTheme(flavor, options);
  return {
    ...compiled,
    name: `catppuccin-${flavor}`
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  compile
});
